--- enlightenment-conf-0.15/e-conf.c.noscrollbar	Fri Mar 12 18:51:57 1999
+++ enlightenment-conf-0.15/e-conf.c	Fri Mar 12 19:26:04 1999
@@ -740,7 +740,7 @@
 	  
 	  kb = ptr->data;
 	  ptr = ptr->next;
-	  if (strcmp(kb->key, "EDIT ME"))
+	  if (strcmp(kb->key, "EDIT"))
 	    {
 	      if (kb->params)
 		g_snprintf(buf2, sizeof(buf2), "%s %i %i %s\n",
@@ -4138,7 +4138,7 @@
   gchar   *line[4];
   
   kb = g_malloc(sizeof(Keybind));
-  kb->key = g_strdup("EDIT ME");
+  kb->key = g_strdup("EDIT");
   kb->modifier = 0;
   kb->id = 1;
   kb->params = NULL;
