--- procmail-3.10/config.h.lockf	Sun Aug 16 21:28:15 1998
+++ procmail-3.10/config.h	Sun Aug 16 21:30:03 1998
@@ -51,8 +51,8 @@
 	delivery mode argument(s) */
 
 /*#define NO_fcntl_LOCK		/* uncomment any of these three if you	     */
-/*#define NO_lockf_LOCK		/* definitely do not want procmail to make   */
-/*#define NO_flock_LOCK		/* use of those kernel-locking methods	     */
+#define NO_lockf_LOCK		/* definitely do not want procmail to make   */
+#define NO_flock_LOCK		/* use of those kernel-locking methods	     */
 
 #define NO_NFS_ATIME_HACK	/* uncomment if you're definitely not using
 				   NFS mounted filesystems and can't afford
