--- procmail-3.10/Makefile.marc	Thu Feb  8 14:43:56 1996
+++ procmail-3.10/Makefile	Thu Feb  8 14:44:22 1996
@@ -61,7 +61,8 @@
 # Only edit below this line if you *think* you know what you are doing #
 ########################################################################
 
-#LOCKINGTEST=100	# Uncomment (and change) if you think you know
+# As per Arnt Gulbrandsen <agulbra@troll.no>, we set this to 000
+LOCKINGTEST=000	        # Uncomment (and change) if you think you know
 #			it better than the autoconf lockingtests.
 #			This will cause the lockingtests to be hotwired.
 #			100	to enable fcntl()
