--- libgr-2.0.13/ppm/bmp.h.no24	Mon Apr  7 00:50:36 1997
+++ libgr-2.0.13/ppm/bmp.h	Thu Mar 11 08:36:10 1999
@@ -1,5 +1,5 @@
 /*\
- * $Id: bmp.h,v 1.1 1997/04/06 23:50:36 neal Exp $
+ * $Id: bmp.h,v 1.3 1992/11/24 19:39:56 dws Exp dws $
  * 
  * bmp.h - routines to calculate sizes of parts of BMP files
  *
@@ -18,9 +18,6 @@
  * without express or implied warranty.
  * 
  * $Log: bmp.h,v $
- * Revision 1.1  1997/04/06 23:50:36  neal
- * From netpbm.
- *
  * Revision 1.3  1992/11/24  19:39:56  dws
  * Added copyright.
  *
@@ -111,7 +108,12 @@
 		return 0;
 	}
 
-	return (1 << bitcount) * lenrgb;
+	if (bitcount==24) {
+		/* samjam@bigfoot.com: no RGB table for 23 bit bmp's */
+		return 0;
+	} else {
+		return (1 << bitcount) * lenrgb;
+	}
 }
 
 /*
