--- cleanfeed-0.95.7b/cleanfeed~	Thu Sep  3 15:48:14 1998
+++ cleanfeed-0.95.7b/cleanfeed	Thu Sep  3 15:53:49 1998
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # Cleanfeed
 #  Version 0.95.7b  3 September 1998
@@ -26,7 +26,7 @@
 # directory where the cleanfeed.conf and cleanfeed.local files live
 # set this to undef to not use the external file
 
-$config_dir = '/usr/news/bin/control';
+$config_dir = '/etc/news';
 
 
 # Everything below here can be set in cleanfeed.conf
