--- AfterStep-APPS-990204/ascp-1.0/Makefile.in.orig	Fri Feb  5 17:22:39 1999
+++ AfterStep-APPS-990204/ascp-1.0/Makefile.in	Fri Feb  5 17:23:04 1999
@@ -12,11 +12,11 @@
 install:
 	@for I in ${subdirs}; do (cd $$I; ${MAKE} install || exit 1); done
 
-	if [ -d @datadir@/afterstep/ascp ] ; then \
-		$(RMF) -r @datadir@/afterstep/ascp ; \
+	if [ -d $(DESTDIR)/@datadir@/afterstep/ascp ] ; then \
+		$(RMF) -r $(DESTDIR)/@datadir@/afterstep/ascp ; \
 	fi; \
-	$(MKDIR) -p @datadir@/afterstep/ascp ; \
-	$(CP) -R ascp @datadir@/afterstep ;
+	$(MKDIR) -p $(DESTDIR)/@datadir@/afterstep/ascp ; \
+	$(CP) -R ascp $(DESTDIR)/@datadir@/afterstep ;
 
 clean:
 	@for I in ${subdirs}; do (cd $$I; ${MAKE} clean || exit 1); done
--- AfterStep-APPS-990329/ascp-1.0/src/gui.c.compile	Mon Mar 29 19:45:22 1999
+++ AfterStep-APPS-990329/ascp-1.0/src/gui.c	Mon Mar 29 19:45:31 1999
@@ -353,7 +353,7 @@
 		{
 			gtk_label_set(GTK_LABEL(Label2), label2val);
 		}
-		if(pctprogress >= 0 )
+		if(pctprogress >= 0.0 && pctprogress <= 1.0  )
 		{
 			gtk_progress_bar_update(GTK_PROGRESS_BAR(ProgressBar), pctprogress);
 		}
