diff -uNr xgalaga-2.0.34/Makefile.in xgalaga-2.0.34.acme/Makefile.in
--- xgalaga-2.0.34/Makefile.in	Mon May 11 03:52:59 1998
+++ xgalaga-2.0.34.acme/Makefile.in	Thu Sep 10 18:15:06 1998
@@ -43,16 +43,14 @@
 	$(CC) $(LDFLAGS) -o $(SOUNDSRV) $(SOUNDOBJS) $(SOUNDLIBS)
 
 install: all
-	./mkinstalldirs $(bindir)
-	./mkinstalldirs $(prefix)
-	$(INSTALL) $(TARGET) $(bindir)/$(TARGET)
-	$(INSTALL) $(SOUNDSRV) $(prefix)/$(SOUNDSRV)
-	./mkinstalldirs $(prefix)/sounds
-	./mkinstalldirs $(prefix)/levels
-	for i in levels/*.xgl; do $(INSTALL) $$i $(prefix)/$$i ; done
-	for i in sounds/*.raw; do $(INSTALL) $$i $(prefix)/$$i ; done
-	touch $(prefix)/scores
-	chmod 666 $(prefix)/scores
+	./mkinstalldirs $(DESTDIR)$(bindir)
+	./mkinstalldirs $(DESTDIR)$(prefix)
+	$(INSTALL) $(TARGET) $(DESTDIR)$(bindir)/$(TARGET)
+	$(INSTALL) $(SOUNDSRV) $(DESTDIR)$(prefix)/$(SOUNDSRV)
+	./mkinstalldirs $(DESTDIR)$(prefix)/sounds
+	./mkinstalldirs $(DESTDIR)$(prefix)/levels
+	for i in levels/*.xgl; do $(INSTALL) $$i $(DESTDIR)$(prefix)/$$i ; done
+	for i in sounds/*.raw; do $(INSTALL) $$i $(DESTDIR)$(prefix)/$$i ; done
 
 subdirs:
 	for i in $(SUBDIRS); do cd $$i ; make ; done
@@ -78,4 +76,4 @@
 	cd libsprite;$(MAKE) distclean
 
 depend:
-	makedepend $(CFLAGS) $(SRCS)
\ No newline at end of file
+	makedepend $(CFLAGS) $(SRCS)
diff -uNr xgalaga-2.0.34/configure xgalaga-2.0.34.acme/configure
--- xgalaga-2.0.34/configure	Mon May 11 04:02:12 1998
+++ xgalaga-2.0.34.acme/configure	Thu Sep 10 18:06:51 1998
@@ -9,9 +9,9 @@
 
 # Defaults:
 ac_help=
-ac_default_prefix=/usr/local
+ac_default_prefix=/usr/
 # Any additions from configure.in:
-ac_default_prefix=/usr/local/lib/xgalaga
+ac_default_prefix=/usr/lib/xgalaga
 ac_help="$ac_help
   --with-xpm-lib          Directory containing libXpm (may be autodetected)"
 ac_help="$ac_help
@@ -1894,7 +1894,7 @@
 
 
 cat >> confdefs.h <<EOF
-#define SCOREFILE "$prefix/scores"
+#define SCOREFILE "/var/lib/games/xgalscores"
 EOF
 
 cat >> confdefs.h <<EOF
diff -uNr xgalaga-2.0.34/configure.in xgalaga-2.0.34.acme/configure.in
--- xgalaga-2.0.34/configure.in	Mon May 11 03:37:19 1998
+++ xgalaga-2.0.34.acme/configure.in	Thu Sep 10 18:05:53 1998
@@ -1,9 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(data.h)
 AC_CONFIG_HEADER(config.h) 
-AC_PREFIX_DEFAULT(/usr/local/lib/xgalaga)
+AC_PREFIX_DEFAULT(/usr/lib/xgalaga)
 
-test "$prefix" = NONE && prefix=/usr/local/lib/xgalaga
+test "$prefix" = NONE && prefix=/usr/lib/xgalaga
 test "$exec_prefix" = NONE && exec_prefix=/usr/local/bin
 
 AC_CANONICAL_HOST
diff -uNr xgalaga-2.0.34/libsprite/Makefile.in xgalaga-2.0.34.acme/libsprite/Makefile.in
--- xgalaga-2.0.34/libsprite/Makefile.in	Sun Apr 12 03:03:22 1998
+++ xgalaga-2.0.34.acme/libsprite/Makefile.in	Thu Sep 10 17:53:47 1998
@@ -35,8 +35,8 @@
 all: $(TARGET)
 
 install: all
-	mkinstalldirs $(libdir)
-	$(INSTALL) $(TARGET) $(libdir)/$(TARGET)
+	../mkinstalldirs $(DESTDIR)$(libdir)
+	$(INSTALL) $(TARGET) $(DESTDIR)$(libdir)/$(TARGET)
 
 $(TARGET): $(OBJS)
 	$(RM) $(TARGET)
