diff -urbBN sudo.v1.5.4.orig/options.h sudo.v1.5.4/options.h
--- sudo.v1.5.4.orig/options.h	Wed Nov 13 22:50:07 1996
+++ sudo.v1.5.4/options.h	Fri Nov 22 01:28:22 1996
@@ -33,36 +33,36 @@
 
 /*#define FQDN			/* expect fully qualified hosts in sudoers */
 #define LOGGING SLOG_SYSLOG	/* log via SLOG_SYSLOG, SLOG_FILE, SLOG_BOTH */
-#define LOGFAC LOG_LOCAL2	/* syslog facility for sudo to use */
+#define LOGFAC LOG_AUTH		/* syslog facility for sudo to use */
 #define MAXLOGFILELEN 80	/* max chars per log line (for line wrapping) */
 /*#define NO_ROOT_SUDO		/* root is not allowed to use sudo */
 #define ALERTMAIL "root"	/* user that gets sudo mail */
 #define SEND_MAIL_WHEN_NO_USER	/* send mail when user not in sudoers file */
 /*#define SEND_MAIL_WHEN_NOT_OK	/* send mail if no permissions to run command */
 /*#define EXEMPTGROUP "sudo"	/* no passwd needed for users in this group */
-/*#define ENV_EDITOR		/* visudo honors EDITOR and VISUAL envars */
+#define ENV_EDITOR		/* visudo honors EDITOR and VISUAL envars */
 #define SHORT_MESSAGE		/* short sudo message, no copyright printed */
 /*#define NO_MESSAGE		/* no sudo "lecture" message */
 #define TIMEOUT 5		/* minutes before sudo asks for passwd again */
 #define PASSWORD_TIMEOUT 5	/* passwd prompt timeout (in minutes) */
 #define TRIES_FOR_PASSWORD 3	/* number of tries to enter passwd correctly */
-/*#define USE_INSULTS		/* insult the user for incorrect passwords */
-/*#define CLASSIC_INSULTS	/* sudo "classic" insults--need USE_INSULTS */
-/*#define HAL_INSULTS		/* 2001-like insults--must define USE_INSULTS */
-/*#define GOONS_INSULTS		/* Goon Show insults--must define USE_INSULTS */
-/*#define CSOPS_INSULTS		/* CSOps insults--must define USE_INSULTS */
+#define USE_INSULTS		/* insult the user for incorrect passwords */
+#define CLASSIC_INSULTS	/* sudo "classic" insults--need USE_INSULTS */
+#define HAL_INSULTS		/* 2001-like insults--must define USE_INSULTS */
+#define GOONS_INSULTS		/* Goon Show insults--must define USE_INSULTS */
+#define CSOPS_INSULTS		/* CSOps insults--must define USE_INSULTS */
 #define EDITOR _PATH_VI		/* default editor to use */
 #define MAILER _PATH_SENDMAIL	/* what mailer to use */
 #define UMASK 0022		/* umask that the root-run prog should use */
 #define INCORRECT_PASSWORD "Sorry, try again." /* message for bad passwd */
 #define MAILSUBJECT "*** SECURITY information for %h ***" /* mail subject */
 #define PASSPROMPT "Password:"	/* default password prompt */
-/*#define IGNORE_DOT_PATH	/* ignore '.' in $PATH if it exists */
+#define IGNORE_DOT_PATH	/* ignore '.' in $PATH if it exists */
 /*#define SECURE_PATH	"/bin:/usr/ucb:/usr/bin:/usr/etc:/etc" /* secure path */
 /*#define USE_EXECV		/* use execv() instead of execvp() */
 /*#define SHELL_IF_NO_ARGS	/* if sudo is given no arguments run a shell */
 /*#define SHELL_SETS_HOME	/* -s sets $HOME to runas user's homedir */
-/*#define USE_TTY_TICKETS	/* have a different ticket file for each tty */
+#define USE_TTY_TICKETS	/* have a different ticket file for each tty */
 /*#define OTP_ONLY		/* validate user via OTP (skey/opie) only */
 /*#define LONG_OTP_PROMPT	/* use a two line OTP (skey/opie) prompt */
 #define FAST_MATCH		/* command check fails if basenames not same */
