--- wu-ftpd-2.4.2-vr17/src/pathnames.h.linux	Mon Mar 29 20:31:41 1999
+++ wu-ftpd-2.4.2-vr17/src/pathnames.h	Fri Apr 16 03:52:18 1999
@@ -42,7 +42,7 @@
 #define LASTLOG_DIR     "/usr/adm"
 #endif
 
-#define _PATH_EXECPATH  "/bin/ftp-exec"
+#define _PATH_EXECPATH  "/usr/bin/ftp-exec"
 
 #ifdef USE_ETC
 #define _PATH_FTPUSERS  "/etc/ftpusers"
--- wu-ftpd-2.4.2-vr17/build.linux	Thu Mar  4 01:45:57 1999
+++ wu-ftpd-2.4.2-vr17/build	Fri Apr 16 03:52:22 1999
@@ -106,20 +106,20 @@
         echo ''
         echo "Making support library."
         cd ../support
-        make $makeargs $makeopts libsupport.a
+        make "RPM_OPT_FLAGS=$RPM_OPT_FLAGS" $makeargs $makeopts libsupport.a
         echo ''
         echo "Making ftpd."
         cd ../src
-        make $makeargs $makeopts  ftpd
+        make "RPM_OPT_FLAGS=$RPM_OPT_FLAGS" $makeargs $makeopts  ftpd
         echo ''
         echo "Making ftpcount."
-        make $makeargs $makeopts  ftpcount
+        make "RPM_OPT_FLAGS=$RPM_OPT_FLAGS" $makeargs $makeopts  ftpcount
         echo ''
         echo "Making ftpshut".
-        make $makeargs $makeopts  ftpshut
+        make "RPM_OPT_FLAGS=$RPM_OPT_FLAGS" $makeargs $makeopts  ftpshut
         echo ''
         echo "Making ckconfig."
-        make $makeargs $makeopts  ckconfig
+        make "RPM_OPT_FLAGS=$RPM_OPT_FLAGS" $makeargs $makeopts  ckconfig
         cd ..
         if [ ! -d bin ] ;  then    mkdir bin;        fi
         cd bin
--- wu-ftpd-2.4.2-vr17/rhsconfig/ftpaccess.rhsconfig	Fri Apr 16 03:57:21 1999
+++ wu-ftpd-2.4.2-vr17/rhsconfig/ftpaccess	Fri Apr 16 03:57:21 1999
@@ -0,0 +1,24 @@
+class   all   real,guest,anonymous  *
+
+email root@localhost
+
+loginfails 5
+
+readme  README*    login
+readme  README*    cwd=*
+
+message /welcome.msg            login
+message .message                cwd=*
+
+compress        yes             all
+tar             yes             all
+chmod		no		guest,anonymous
+delete		no		guest,anonymous
+overwrite	no		guest,anonymous
+rename		no		guest,anonymous
+
+log transfers anonymous,real inbound,outbound
+
+shutdown /etc/shutmsg
+
+passwd-check rfc822 warn
--- wu-ftpd-2.4.2-vr17/rhsconfig/ftpconversions.rhsconfig	Fri Apr 16 03:57:21 1999
+++ wu-ftpd-2.4.2-vr17/rhsconfig/ftpconversions	Fri Apr 16 03:57:21 1999
@@ -0,0 +1,7 @@
+ :.Z:  :  :/bin/compress -d -c %s:T_REG|T_ASCII:O_UNCOMPRESS:UNCOMPRESS
+ :   : :.Z:/bin/compress -c %s:T_REG:O_COMPRESS:COMPRESS
+ :.gz: :  :/bin/gzip -cd %s:T_REG|T_ASCII:O_UNCOMPRESS:GUNZIP
+ :   : :.gz:/bin/gzip -9 -c %s:T_REG:O_COMPRESS:GZIP
+ :   : :.tar:/bin/tar -c -f - %s:T_REG|T_DIR:O_TAR:TAR
+ :   : :.tar.Z:/bin/tar -c -Z -f - %s:T_REG|T_DIR:O_COMPRESS|O_TAR:TAR+COMPRESS
+ :   : :.tar.gz:/bin/tar -c -z -f - %s:T_REG|T_DIR:O_COMPRESS|O_TAR:TAR+GZIP
--- wu-ftpd-2.4.2-vr17/rhsconfig/ftpgroups.rhsconfig	Fri Apr 16 03:57:21 1999
+++ wu-ftpd-2.4.2-vr17/rhsconfig/ftpgroups	Fri Apr 16 03:57:21 1999
@@ -0,0 +1 @@
+# test:ENCRYPTED PASSWORD HERE:archive
--- wu-ftpd-2.4.2-vr17/rhsconfig/ftphosts.rhsconfig	Fri Apr 16 03:57:21 1999
+++ wu-ftpd-2.4.2-vr17/rhsconfig/ftphosts	Fri Apr 16 03:57:21 1999
@@ -0,0 +1,5 @@
+# Example host access file
+#
+# Everything after a '#' is treated as comment,
+# empty lines are ignored
+
--- wu-ftpd-2.4.2-vr17/rhsconfig/ftpusers.rhsconfig	Fri Apr 16 03:57:21 1999
+++ wu-ftpd-2.4.2-vr17/rhsconfig/ftpusers	Fri Apr 16 03:57:21 1999
@@ -0,0 +1,14 @@
+root
+bin
+daemon
+adm
+lp
+sync
+shutdown
+halt
+mail
+news
+uucp
+operator
+games
+nobody
