--- sliplogin-2.1.1/sliplogin.h~	Wed May  1 12:26:09 1996
+++ sliplogin-2.1.1/sliplogin.h	Wed Dec 30 16:58:32 1998
@@ -80,3 +80,41 @@
     char option[3][16];
 };
 
+
+/*
+ *      Network device statistics. Akin to the 2.0 ether stats but
+ *      with byte counters.
+ */
+
+struct net_device_stats
+{
+    unsigned long   rx_packets;             /* total packets received       */
+    unsigned long   tx_packets;             /* total packets transmitted    */
+    unsigned long   rx_bytes;               /* total bytes received         */
+    unsigned long   tx_bytes;               /* total bytes transmitted      */
+    unsigned long   rx_errors;              /* bad packets received         */
+    unsigned long   tx_errors;              /* packet transmit problems     */
+    unsigned long   rx_dropped;             /* no space in linux buffers    */
+    unsigned long   tx_dropped;             /* no space available in linux  */
+    unsigned long   multicast;              /* multicast packets received   */
+            unsigned long   collisions;
+
+    /* detailed rx_errors: */
+            unsigned long   rx_length_errors;
+    unsigned long   rx_over_errors;         /* receiver ring buff overflow  */
+    unsigned long   rx_crc_errors;          /* recved pkt with crc error    */
+    unsigned long   rx_frame_errors;        /* recv'd frame alignment error */
+    unsigned long   rx_fifo_errors;         /* recv'r fifo overrun          */
+    unsigned long   rx_missed_errors;       /* receiver missed packet       */
+
+    /* detailed tx_errors */
+            unsigned long   tx_aborted_errors;
+            unsigned long   tx_carrier_errors;
+            unsigned long   tx_fifo_errors;
+            unsigned long   tx_heartbeat_errors;
+            unsigned long   tx_window_errors;
+
+    /* for cslip etc */
+            unsigned long   rx_compressed;
+            unsigned long   tx_compressed;
+};
--- sliplogin-2.1.1/statistic.c~	Sat May  4 12:27:15 1996
+++ sliplogin-2.1.1/statistic.c	Wed Dec 30 16:59:40 1998
@@ -1,4 +1,6 @@
 
+#include "sliplogin.h"
+
 #ifdef linux
 
 #include <stdio.h>
