--- ext2ed-0.1/ext2ed.h.compat21	Tue Dec 22 12:37:16 1998
+++ ext2ed-0.1/ext2ed.h	Tue Dec 22 12:37:50 1998
@@ -35,7 +35,7 @@
 #define DEBUG						/* Activate self-sanity checks */
 
 #include <linux/ext2_fs.h>				/* Main kernel ext2 include file */
-#include <linux/stat.h>
+#include <sys/stat.h>
 
 #ifdef OLD_NCURSES					/* The ncurses interface */
 	#include <ncurses/ncurses.h>
@@ -435,4 +435,4 @@
 extern void redraw_all (void);
 extern void close_windows (void);
 
-#endif /* EXT2ED_EDITOR_H */
\ No newline at end of file
+#endif /* EXT2ED_EDITOR_H */
