--- control-center-1.0.2/capplets/sound-properties/sound-properties.c.nosound	Wed Mar 10 18:05:17 1999
+++ control-center-1.0.2/capplets/sound-properties/sound-properties.c	Wed Mar 10 18:06:35 1999
@@ -131,7 +131,7 @@
 #ifdef HAVE_ESD
         if(token) {
             if(gnome_sound_connection < 0
-               && gnome_config_get_bool("/sound/system/settings/start_esd=true")) {
+               && gnome_config_get_bool("/sound/system/settings/start_esd=false")) {
                 int esdpid;
                 static const char *esd_cmdline[] = {"esd", "-nobeeps", NULL};
                 char *tmpargv[3];
@@ -244,7 +244,7 @@
                         FALSE, FALSE, 0);
                         
     gtk_toggle_button_set_state(GTK_TOGGLE_BUTTON(retval->enable_esd_startup),
-                                gnome_config_get_bool("/sound/system/settings/start_esd=true"));
+                                gnome_config_get_bool("/sound/system/settings/start_esd=false"));
 
     retval->enable_sound_events =
         gtk_check_button_new_with_label(_("Sounds for events"));
@@ -636,7 +636,7 @@
 ui_do_revert(GtkWidget *w, SoundProps *props)
 {
     gtk_toggle_button_set_state(GTK_TOGGLE_BUTTON(props->enable_esd_startup),
-                                gnome_config_get_bool("/sound/system/settings/start_esd=true"));
+                                gnome_config_get_bool("/sound/system/settings/start_esd=false"));
 
     gtk_toggle_button_set_state(GTK_TOGGLE_BUTTON(props->enable_sound_events),
                                 gnome_config_get_bool("/sound/system/settings/event_sounds=true"));
