--- efax08a/Makefile.orig	Sun Sep 15 12:30:13 1996
+++ efax08a/Makefile	Mon Nov 25 18:50:13 1996
@@ -10,8 +10,8 @@
 # For SunOS: add -DNO_STRERROR
 #
 
-CFLAGS=-O
-LDFLAGS=
+CFLAGS=$(RPM_OPT_FLAGS)
+LDFLAGS=-s
 
 # Change the following to the destination directories for
 # binaries and man pages. Probably /usr/bin and /usr/man on
--- efax08a/fax.orig	Mon Aug 26 03:41:29 1996
+++ efax08a/fax	Mon Nov 25 18:51:56 1996
@@ -11,9 +11,9 @@
 # The names of the fax script, efax and efix including full path
 # if necessary.
 
-FAX=fax
-EFAX=efax
-EFIX=efix
+FAX=/usr/bin/fax
+EFAX=/usr/bin/efax
+EFIX=/usr/bin/efix
 
 # The device to which the fax modem is connected (e.g. ttya for
 # /dev/ttya).  Use a dial-out (cua) device if available.  If
@@ -22,7 +22,7 @@
 # /dev/modem is a link to /dev/cua1, then getty, uucp, kermit,
 # pppd, dip, etc. must *all* use either /dev/modem or /dev/cua1.
 
-DEV=cua1
+DEV=modem
 
 # The type of fax modem commands to use.  Un-comment one of the
 # following lines depending on your modem.
@@ -34,11 +34,11 @@
 # Your fax number in international format, 20 characters maximum.
 # Use only digits, spaces, and the "+" character.
 
-FROM="+1 800 555 5555"
+FROM="+0 000 000 0000"
 
 # Your name as it should appear on the page header.
 
-NAME="Put Your Name Here"
+NAME="from a Red Hat Linux system"
 
 # The preferred page size for creating and printing faxes.
 # Allowed values are "letter", "legal", and "a4".
@@ -50,8 +50,8 @@
 # The type of printer. Use 'pcl' for HP-PCL or 'ps' for
 # Postscript.  See definition of PRINT (below) for more options.
 
-PRTYPE=pcl				# HP-PCL (e.g. HP LaserJet) 
-# PRTYPE=ps				# Postscript (e.g. Apple LaserWriter)
+# PRTYPE=pcl				# HP-PCL (e.g. HP LaserJet) 
+PRTYPE=ps				# Postscript (e.g. Apple LaserWriter)
 
 # The command to print image files from standard input.  Typically
 # this is "lpr" or "lp".
@@ -68,7 +68,7 @@
 # The name of the Ghostscript executable including full path if
 # necessary.  Only required if faxing Postscript files.
 
-GS=gs
+GS=/usr/bin/gs
 
 # Dial string prefix and suffix such as T for tone dialing, P for
 # pulse dialing, 9 to get an external line, commas for delays or
@@ -82,7 +82,7 @@
 # conventions. Protect with single quotes for delayed evaluation.
 # Add a leading '#' to the file name to use HDB (ASCII) format.
 
-LOCK='-x /usr/spool/uucp/LCK..$DEV'		# "normal" systems
+LOCK='-x /var/lock/LCK..$DEV'		# "normal" systems
 # LOCK=''					# no lock file
 # LOCK='-x /usr/spool/locks/LK.047.040.011'	# SysV style names
 # LOCK='-x #/usr/spool/uucp/LCK..$DEV'		# HDB format (ASCII pid)
