--- traceroute-1.4a5/linux-include/netinet/ip_var.h.ink	Wed Jan  1 03:50:24 1997
+++ traceroute-1.4a5/linux-include/netinet/ip_var.h	Wed Dec  2 13:34:28 1998
@@ -39,7 +39,7 @@
  * Overlay for ip header used by other protocols (tcp, udp).
  */
 struct ipovly {
-	caddr_t	ih_next, ih_prev;	/* for protocol sequence q's */
+	u_int32_t ih_next, ih_prev;	/* for protocol sequence q's */
 	u_char	ih_x1;			/* (unused) */
 	u_char	ih_pr;			/* protocol */
 	short	ih_len;			/* protocol length */

