--- ncpfs-2.2.0.12/ipx-1.0/Makefile.paths	Mon May  4 09:13:58 1998
+++ ncpfs-2.2.0.12/ipx-1.0/Makefile	Tue Feb 23 01:31:41 1999
@@ -17,6 +17,7 @@
 	rm -f .depend
 
 install: $(UTILS)
+	mkdir -p $(BINDIR)
 	for i in $(UTILS); \
 	do \
 		install $$i $(BINDIR); \
--- ncpfs-2.2.0.12/sutil/Makefile.paths	Fri Dec  4 12:31:36 1998
+++ ncpfs-2.2.0.12/sutil/Makefile	Tue Feb 23 01:31:41 1999
@@ -21,6 +21,7 @@
 all: $(UTILS)
 
 install: all
+	mkdir -p $(BINDIR)
 	for i in $(UTILS); \
 	do install $$i -m 4755 $(BINDIR); done
 	ln -sf $(BINDIR)/ncpmount /sbin/mount.ncp 
--- ncpfs-2.2.0.12/util/Makefile.paths	Fri Jul 17 14:03:31 1998
+++ ncpfs-2.2.0.12/util/Makefile	Tue Feb 23 01:31:41 1999
@@ -35,6 +35,7 @@
 all: $(UTILS) ncptest # ipx_probe
 
 install: all
+	mkdir -p $(BINDIR) $(SBINDIR)
 	for i in $(USERUTILS); \
 	do install $$i -m 755 $(BINDIR); done
 	for i in $(SBINUTILS); \
--- ncpfs-2.2.0.12/lib-shared/Makefile.paths	Mon May 11 13:31:34 1998
+++ ncpfs-2.2.0.12/lib-shared/Makefile	Tue Feb 23 01:31:41 1999
@@ -16,6 +16,7 @@
 
 install: all
 	rm -f $(LIBSODIR)/$(NCPLIB)
+	mkdir -p $(LIBSODIR)
 	install $(NCPLIB_BIN) $(LIBSODIR)
 	ldconfig
 	ln -sf $(NCPLIB_SONAME) $(LIBSODIR)/$(NCPLIB)
--- ncpfs-2.2.0.12/Makeinit.paths	Fri Feb 12 14:32:14 1999
+++ ncpfs-2.2.0.12/Makeinit	Tue Feb 23 01:32:44 1999
@@ -36,12 +36,12 @@
 MOUNT3 = 1
 
 # ########################################################
-BINDIR = /usr/bin
-SBINDIR = /sbin
+BINDIR = $(DESTDIR)/usr/bin
+SBINDIR = $(DESTDIR)/sbin
 #LIBSODIR = /usr/lib
-LIBSODIR = /lib
-LIBADIR = /usr/lib
-MANDIR = /usr/man
+LIBSODIR = $(DESTDIR)/lib
+LIBADIR = $(DESTDIR)/usr/lib
+MANDIR = $(DESTDIR)/usr/man
 
 CC = gcc
 
