--- xosview-1.6.1.a/config/Makefile.linux.in.sparc	Fri Apr  3 12:17:16 1998
+++ xosview-1.6.1.a/config/Makefile.linux.in	Tue Jun 16 16:50:35 1998
@@ -16,7 +16,6 @@
 pagemeter.o \
 netmeter.o \
 intmeter.o \
-serialmeter.o \
 loadmeter.o \
 btrymeter.o \
 MeterMaker.o
--- xosview-1.6.1.a/linux/MeterMaker.cc.sparc	Tue Jun 16 16:52:26 1998
+++ xosview-1.6.1.a/linux/MeterMaker.cc	Tue Jun 16 16:53:32 1998
@@ -50,11 +50,13 @@
   if (_xos->isResourceTrue("net"))
     push(new NetMeter(_xos, atof(_xos->getResource("netBandwidth"))));
 
+#if 0
   // check for the serial meters.
   for (int i = 0 ; i < SerialMeter::numDevices() ; i++)
     if (strcmp(_xos->getResource(
       SerialMeter::getResourceName((SerialMeter::Device)i)), "False"))
       push(new SerialMeter(_xos, (SerialMeter::Device)i));
+#endif
 
   // check for the interrupt meter
   if (_xos->isResourceTrue("interrupts")) {
