diff -cNr /tmp/cpio-2.3/copyout.c cpio-2.3/copyout.c
*** /tmp/cpio-2.3/copyout.c	Wed Apr 28 15:33:12 1993
--- cpio-2.3/copyout.c	Wed Jun 22 23:16:23 1994
***************
*** 452,464 ****
  	      {
  		char *link_name = (char *) xmalloc (file_stat.st_size + 1);
  
! 		if (readlink (input_name.ds_string, link_name,
! 			      file_stat.st_size) < 0)
  		  {
  		    error (0, errno, "%s", input_name.ds_string);
  		    free (link_name);
  		    continue;
  		  }
  		if (archive_format == arf_tar || archive_format == arf_ustar)
  		  {
  		    if (file_stat.st_size + 1 > 100)
--- 452,466 ----
  	      {
  		char *link_name = (char *) xmalloc (file_stat.st_size + 1);
  
! 		/* Reset st_size to handle imprecise lstat st_sizes */
! 		if ((file_stat.st_size = readlink (input_name.ds_string, link_name,
! 			      file_stat.st_size)) < 0)
  		  {
  		    error (0, errno, "%s", input_name.ds_string);
  		    free (link_name);
  		    continue;
  		  }
+ 		file_hdr.c_filesize = file_stat.st_size;
  		if (archive_format == arf_tar || archive_format == arf_ustar)
  		  {
  		    if (file_stat.st_size + 1 > 100)
