--- X11R6-contrib-3.1.2/contrib/programs/Imakefile.djb	Thu Aug 29 16:59:18 1996
+++ X11R6-contrib-3.1.2/contrib/programs/Imakefile	Thu Aug 29 16:59:33 1996
@@ -3,6 +3,7 @@
 
 #define IHaveSubdirs
 #define PassCDebugFlags CDEBUGFLAGS="$(CDEBUGFLAGS)"
+#define XF86Contrib
 
 #if BuildPhigs
 PHIGSDIRS = auto_box beach_ball gpc
--- X11R6-contrib-3.1.2/contrib/Imakefile.djb	Thu Aug 29 16:59:11 1996
+++ X11R6-contrib-3.1.2/contrib/Imakefile	Thu Aug 29 16:51:33 1996
@@ -3,6 +3,7 @@
 
 #define IHaveSubdirs
 #define PassCDebugFlags CDEBUGFLAGS="$(CDEBUGFLAGS)"
+#define XF86Contrib
 
 #ifdef XF86Contrib
 SUBDIRS = programs
