--- emacs-20.3/lisp/uncompress.el.ufix	Thu Jun 11 18:52:13 1998
+++ emacs-20.3/lisp/uncompress.el	Wed Sep 30 21:38:50 1998
@@ -72,14 +72,8 @@
           (set-visited-file-name
            (concat (substring buffer-file-name 0 (match-beginning 0)) ".tar")))))
   (message "Uncompressing...")
-  (let ((buffer-read-only nil)
-	(coding-system-for-write 'no-conversion)
-	(coding-system-for-read
-	 (find-operation-coding-system
-	  'insert-file-contents
-	  buffer-file-name t)))
+  (let ((buffer-read-only nil))
     (shell-command-on-region (point-min) (point-max) uncompress-program t))
-  (goto-char (point-min))
   (message "Uncompressing...done")
   (set-buffer-modified-p nil)
   (make-local-variable 'write-file-hooks)
@@ -104,9 +98,7 @@
 	(progn
 	  (insert-file-contents buffer-file-name t)
 	  (goto-char (point-min))
-	  ;; No need for this, because error won't be set to t
-	  ;; if this function returns t.
-	  ;; (setq error nil)
+	  (setq error nil)
 	  t))))
 
 (provide 'uncompress)
