--- trn-3.6/final.c.ewt	Tue Dec 30 23:45:50 1997
+++ trn-3.6/final.c	Tue Dec 30 23:48:14 1997
@@ -222,6 +222,8 @@
 stop_catcher(signo)
 int signo;
 {
+    sigset_t stopmask;
+
     if (!waiting) {
 	bool mouse_flag;	/* if TRUE, mouse tracking was on */
 	mouse_flag = xmouse_status;
@@ -237,11 +239,16 @@
 	if (debug)
 	    write(2,"stop_catcher\n",13);
 #endif
-	sigset(signo,SIG_DFL);	/* enable stop */
-#ifdef HAS_SIGBLOCK
-	sigsetmask(sigblock(0) & ~(1 << (signo-1)));
-#endif
+	signal(signo,SIG_DFL);	/* enable stop */
+
+	sigemptyset(&stopmask);
+	sigaddset(&stopmask, SIGTSTP);
+	sigprocmask(SIG_UNBLOCK, &stopmask, NULL);
+
 	kill(0,signo);		/* and do the stop */
+
+	sigprocmask(SIG_BLOCK, &stopmask, NULL);
+
     	savetty();
 #ifdef MAILCALL
     	mailcount = 0;			/* force recheck */
