--- swatch-2.2/install.pl.redhat	Thu Dec 14 17:17:23 1995
+++ swatch-2.2/install.pl	Wed Apr 29 17:25:52 1998
@@ -12,32 +12,33 @@
 
 # usage: install.pl [full_path_to_perl]
 #
-$ENV{'PATH'} = '/usr/ucb:/usr/bin:/bin:/usr/local/bin' ;
+$ENV{'PATH'} = '/usr/bin:/bin:/usr/local/bin' ;
 $ENV{'IFS'} = '' if $ENV{'IFS'} ne '' ;
 
 
 ##
 ## Default values
 ##
-       $PERL_LIB_DIR = "/usr/local/lib/perl" ;
-       $PERL_BIN_DIR = "/usr/local/bin" ;
-     $SWATCH_BIN_DIR = "/usr/local/etc" ;
-     $SWATCH_MAN_DIR = "/usr/local/man" ;
+       $PERL_LIB_DIR = "/usr/lib/perl5" ;
+       $PERL_BIN_DIR = "/usr/bin" ;
+     $SWATCH_BIN_DIR = "/usr/bin" ;
+     $SWATCH_LIB_DIR = "/usr/lib" ;
+     $SWATCH_MAN_DIR = "/usr/man" ;
 $SWATCH_PROG_MAN_EXT = "8" ;
 $SWATCH_CONF_MAN_EXT = "5" ;
    $SWATCH_DATA_MODE = 444 ;
  $SWATCH_SCRIPT_MODE = 755 ;
               $OWNER = "root" ;
-              $GROUP = "wheel" ;
+              $GROUP = "root" ;
    $SWATCH_PROG_NAME = "swatch" ;
     $SWATCH_PROG_MAN = "swatch.prog.man" ;
     $SWATCH_CONF_MAN = "swatch.conf.man" ;
   $SWATCH_ACTION_LIB = "sw_actions.pl" ;
     $SWATCH_HIST_LIB = "sw_history.pl" ;
 
-                $SED = "/usr/bin/sed" ; 
-               $COPY = "/usr/bin/cp" ;
-	      $CHMOD = "/usr/bin/chmod" ;
+                $SED = "/bin/sed" ; 
+               $COPY = "/bin/cp" ;
+	      $CHMOD = "/bin/chmod" ;
 
 if ( ! -f $SED ) {
     $SED = `which sed` ;
@@ -58,8 +59,12 @@
 } else {
     $PERL_BIN_DIR = $ARGV[0] ;
 }
+$PERL_BIN_DIR = "/usr/bin/perl";
 $PERL_BIN_DIR = substr($PERL_BIN_DIR, 0, rindex($PERL_BIN_DIR, '/')) ;
 
+&do_the_install();
+exit 0;
+
 $ready = 0 ;
 while (!$ready) {
     &get_some_answers() ;
@@ -147,9 +152,9 @@
     local($uid, $gid, $mode, $src_file, $dest_file) = @_ ;
 
     print "Installing $src_file..." ;
-    system("$COPY $src_file $dest_file") ;
+    system("$COPY -v $src_file $dest_file") ;
     chown($uid, $gid, $dest_file) ;
-    system("$CHMOD $mode $dest_file") ;
+    system("$CHMOD -v $mode $dest_file") ;
     print "done.\n" ;
 }
 
--- swatch-2.2/swatch.pl.redhat	Thu Dec 14 15:01:08 1995
+++ swatch-2.2/swatch.pl	Wed Apr 29 17:23:38 1998
@@ -44,7 +44,7 @@
 
 # Some defaults
 $PERL                   = 'PERL_BIN_DIR/perl';
-$TAIL			= '/usr/ucb/tail -f';
+$TAIL			= '/usr/bin/tail -f';
 $DEF_INPUT		= "$TAIL /var/log/syslog";
 $Pipe			= 1;
 $ConfigFile		= "$ENV{'HOME'}/.swatchrc";
