--- xevil-1.5/Makefile.orig	Mon Jun  3 20:47:02 1996
+++ xevil-1.5/Makefile	Thu Oct  9 14:35:14 1997
@@ -8,7 +8,7 @@
 #### Defines that are applicable to makes on all architectures
 SHELL		=	/bin/sh
 TARGETS		=	xevil #xshow
-DEBUG_OPT	=	-O #-g#  #-DPRINT_ERRORS  
+DEBUG_OPT	=	-fno-for-scope $(RPM_OPT_FLAGS) #-g#  #-DPRINT_ERRORS  
 LINK_OPT	=	-O
 
 #SRC_DIR and WORK_DIR are only used for `make workdir`
@@ -124,7 +124,7 @@
 
 
 
-i386-linux:
+linux:
 	@$(MAKE) CC="g++" CFLAGS="-DUSE_RANDOM -DPROTECTED_IS_PUBLIC" \
 INCL_DIRS="" LIBS_DIRS="-L/usr/X11R6/lib" LIBS="-lX11 -lm" $(TARGETS)
 
