--- teTeX-1.0/texk/kpathsea/withenable.ac.acme	Wed Jun  9 16:52:10 1999
+++ teTeX-1.0/texk/kpathsea/withenable.ac	Wed Jun  9 16:54:34 1999
@@ -49,6 +49,11 @@
 dnl Check whether we can find a texmf tree.  Look at the most common spots.
 dnl Not a switch, but something that must be included everywhere, even at
 dnl at the top level, so putting it in common.ac is not enough.
+AC_ARG_WITH(texmf-dir,
+[  --with-texmf-dir=DIR        use DIR as the main texmf directory],
+
+texmfmain=$with_texmf_dir,
+[
 AC_MSG_CHECKING(where the main texmf tree is located)
 texmfmain=
 if test "x$datadir" != 'x${prefix}/share'; then
@@ -80,6 +85,8 @@
     fi
   done
   texmfmain=`echo "$texmfmain" | sed 's,//*,/,g'`
+])
+
   if test -n "$texmfmain"; then
     AC_MSG_RESULT("$texmfmain")
   else
@@ -94,3 +101,10 @@
   fi
 fi
 AC_SUBST(texmfmain)
+
+AC_ARG_WITH(fonts-dir,
+[  --with-fonts-dir=DIR        use DIR instead for temporary rendered fonts],
+       vartexfonts=$with_fonts_dir,
+       vartexfonts=/var/tmp/texfonts
+)
+AC_SUBST(vartexfonts)
--- teTeX-0.9/texk/make/paths.mk.config	Fri Feb 13 15:15:30 1998
+++ teTeX-0.9/texk/make/paths.mk	Sat Sep 12 13:23:06 1998
@@ -46,7 +46,7 @@
 texmf = @texmfmain@
 
 # The directory used by varfonts.
-vartexfonts = /var/tmp/texfonts
+vartexfonts = @vartexfonts@
 
 # Regular input files.
 texinputdir = $(texmf)/tex
