--- pdksh-5.2.13/trap.c.compat21	Wed Mar 17 15:38:00 1999
+++ pdksh-5.2.13/trap.c	Wed Mar 17 15:39:55 1999
@@ -30,7 +30,7 @@
 
 	/* Use system description, if available, for unknown signals... */
 	for (i = 0; i < NSIG; i++)
-		if (!sigtraps[i].name && sys_siglist[i][0])
+		if (!sigtraps[i].name && sys_siglist[i] && sys_siglist[i][0])
 			sigtraps[i].mess = sys_siglist[i];
 #endif	/* HAVE_SYS_SIGLIST */
 
