--- vga_cardgames/Makefile.orig	Sun Feb 26 08:19:37 1995
+++ vga_cardgames/Makefile	Tue Dec 12 13:42:56 1995
@@ -9,7 +9,7 @@
 
 # Installation directories
 BINDIR=		/usr/games
-LIBDIR=		/usr/games/lib
+LIBDIR=		/usr/lib/games
 
 # Binary name prefix (to avoid clashes with other programs)
 PREFIX=		vga_
@@ -22,7 +22,7 @@
 
 # These may be changed, if desired
 CC=		gcc
-CCOPTS=		-O2 -m486
+CCOPTS=		$(RPM_OPT_FLAGS)
 
 # ------------------------------------------------------------------- #
 
@@ -30,7 +30,7 @@
 
 CFLAGS=	$(CCOPTS) $(OPTIONS) \
 	-DCARDSFILE=\"$(LIBDIR)/Cards56x80\" \
-	-DVGA16FONT=\"$(LIBDIR)/Vga16font8x16\"
+	-DVGA16FONT=\"$(LIBDIR)/Vga16font8x16.cards\"
 OHHELL=		$(PREFIX)ohhell
 OHHELLOBJ=	ohhell.o ohlinux.o common.o mouse.o key.o vga16.o
 SPIDER=		$(PREFIX)spider
@@ -49,7 +49,7 @@
 	install -o root -m 4711 -s $(KLONDIKE) $(BINDIR)
 	install -o root -m 4711 -s $(SOLITAIRE) $(BINDIR)
 	install -m 444 Cards56x80 $(LIBDIR)
-	install -m 444 Vga16font8x16 $(LIBDIR)
+	install -m 444 Vga16font8x16 $(LIBDIR)/Vga16font8x16.cards
 
 
 $(OHHELL):	$(OHHELLOBJ)
