--- mgetty-1.1.5/voice/Makefile.msf	Tue Apr 22 11:06:56 1997
+++ mgetty-1.1.5/voice/Makefile	Tue Apr 22 11:07:09 1997
@@ -81,11 +81,11 @@
 	@echo "  Installing the voice extensions"
 	@echo "  -------------------------------"
 	@echo ""
-	$(INSTALL) -m 700 vgetty/vgetty $(SBINDIR)
-	$(INSTALL) -m 755 vm/vm $(BINDIR)
+	$(INSTALL) -s -m 700 vgetty/vgetty $(SBINDIR)
+	$(INSTALL) -s -m 755 vm/vm $(BINDIR)
 	for i in $(PVFTOOLS); \
 	do \
-	     $(INSTALL) -m 755 pvftools/$$i $(BINDIR); \
+	     $(INSTALL) -s -m 755 pvftools/$$i $(BINDIR); \
 	done
 	$(INSTALL) -m 644 man/man1/zplay.1 $(MAN1DIR)
 	$(INSTALL) -m 644 man/man1/pvf.1 $(MAN1DIR)
