--- mgetty-1.1.14/doc/mgetty.texi-in.docfix	Thu Apr  2 03:11:50 1998
+++ mgetty-1.1.14/doc/mgetty.texi-in	Tue Mar 23 12:08:18 1999
@@ -316,8 +316,8 @@
 If @code{mgetty} or @code{sendfax} are run "as is", they will use their
 compiled-in defaults from @file{policy.h}.
 
-If the configuration files @file{mgetty.config} (@xref{runtime-mgetty}) and
-@file{sendfax.config} (@xref{runtime-sendfax}) exist (and are readable),
+If the configuration files @file{mgetty.config} (@xref{runtime-mgetty}.) and
+@file{sendfax.config} (@xref{runtime-sendfax}.) exist (and are readable),
 both programs will get their run-time configuration from there. Items not
 specified there will still be taken from the compiled-in defaults. Command
 line switches will always override those settings. The configuration files
@@ -524,7 +524,7 @@
 @item
 Linux:
 
-Use @file{/dev/ttyS*}, @emph{not} @file{/dev/cua*}. @xref{Linux}
+Use @file{/dev/ttyS*}, @emph{not} @file{/dev/cua*}. @xref{Linux}.
 
 @item
 SunOS, FreeBSD, NetBSD:
@@ -926,7 +926,7 @@
 
 This is needed if you have a modem that supports ``caller ID'' detection,
 but needs a special command to get the CID information. Right now, this is
-only needed for some ELINK ISDN adaptors (@xref{Elink-ISDN}), most other
+only needed for some ELINK ISDN adaptors (@xref{Elink-ISDN}.), most other
 CID-capable modems send the CID on their own and don't need this.
 
 Don't forget to set @code{rings} to at least 2, otherwise the CID grabbing
@@ -1196,7 +1196,7 @@
 There are three easy ways to create g3 fax files, either use @code{pbm2g3}
 (included in this package. Do not use @code{pbmtog3} from the pbmplus 
 toolkit. @strong{Note:}
-@xref{pbmtog3}), use GhostScript (GNU Software, can be found on
+@xref{pbmtog3}.), use GhostScript (GNU Software, can be found on
 prep.ai.mit.edu) which contains a ``digifax'' driver that will produce the
 required format, or try Chris Lewis' @samp{hp2pbm} package which will
 convert HP Laserjet print files into g3 fax files (@code{hp2hig3}
@@ -1379,7 +1379,7 @@
 but since most class 2 faxmodems do not implement the command set properly,
 putting a header on a page does not work with them - because of that, I had
 to disable the corresponding code.  @code{faxspool} works around this
-problem by using @code{g3cat} (@xref{Fax-Tools}) to ``paste'' a header line
+problem by using @code{g3cat} (@xref{Fax-Tools}.) to ``paste'' a header line
 on top of each page)
 
 If you specify @samp{-v}, sendfax will output more verbose progress
@@ -1733,12 +1733,12 @@
 Force the modem command set to use. Default is @samp{auto} (auto-detect,
 which may not work on very cheap modems), possible other values are
 @samp{cls2}, for ``class 2 only'' modems, and @samp{c2.0} for ``class 2.0''
-faxmodems. Default is @code{DEFAULT_MODEMTYPE}. @xref{runtime-mgetty}
+faxmodems. Default is @code{DEFAULT_MODEMTYPE}. @xref{runtime-mgetty}.
 
 @item @code{modem-quirks} @var{bitmask}
 
 Same as in @code{mgetty}, this can be used to adapt @code{sendfax} to
-some peculiarities in certain modems. @xref{runtime-mgetty}
+some peculiarities in certain modems. @xref{runtime-mgetty}.
 
 @item @code{max-tries} @var{nnn}
 
@@ -2557,7 +2557,7 @@
 @subsection Tornado / Lightspeed modems
 
 Here is a suggested setting for the default profile for these modems.
-@xref{Common}
+@xref{Common}.
 
 For Lightspeed store profile:
 @example
