--- mgetty-1.1.5/Makefile.msf	Mon Apr 21 12:42:20 1997
+++ mgetty-1.1.5/Makefile	Mon Apr 21 12:48:14 1997
@@ -601,6 +601,13 @@
 				BINDIR=$(BINDIR) SBINDIR=$(SBINDIR) \
 				LIBDIR=$(LIBDIR) CONFDIR=$(CONFDIR)
 #
+# tools
+#
+	cd tools ; $(MAKE) install INSTALL="$(INSTALL)" \
+				BINDIR=$(BINDIR) SBINDIR=$(SBINDIR) \
+				LIBDIR=$(LIBDIR) CONFDIR=$(CONFDIR)
+
+#
 # compatibility
 #
 	if [ ! -z "$(INSTALL_MECHO)" ] ; then \
