--- mgetty-1.1.14/policy.h-dist	Tue Dec 16 12:40:38 1997
+++ mgetty-1.1.14/policy.h	Sat Aug 22 23:38:31 1998
@@ -67,7 +67,7 @@
  * ZyXELs use S40.2=1.
  * If the path doesn't start with "/", it's relative to CONFDIR.
  */
-/* #define CNDFILE "dialin.config" */
+#define CNDFILE "dialin.config"
 
 
 /* If you want to use /etc/gettydefs to set tty flags, define this
@@ -96,7 +96,7 @@
 /* group id that the device is chown()ed to. If not defined, the
  * primary group of "DEVICE_OWNER" is used.
  */
-#define DEVICE_GROUP	"modem"
+#define DEVICE_GROUP	"uucp"
 
 /* access mode for the line while getty has it - it should be accessible
  * by uucp / uucp, but not by others (imagine someone dialing into your
@@ -125,7 +125,7 @@
  *
  * if the directory does not exist, the log file goes to CONSOLE (!)
  */
-#define LOG_PATH "/var/log/mgetty.%s"
+#define LOG_PATH "/var/log/mgetty.log.%s"
 
 /* Default log error level threshold. Possible error levels are
  * L_FATAL, L_ERROR, L_WARN, L_AUDIT, L_MESG, L_NOISE, L_JUNK (see mgetty.h)
@@ -150,7 +150,7 @@
  * mgetty will use the facility "LOG_AUTH", and the priorities
  * LOG_NOTICE, LOG_ERR and LOG_ALERT, respectively.
  */
-/* #define SYSLOG */
+#define SYSLOG
 
 /* Syslog facility
  *
@@ -172,7 +172,7 @@
  * current date and time, respectively.
  * override with "-p <prompt>" switch
  */
-#define LOGIN_PROMPT	"@!login: "
+#define LOGIN_PROMPT	"@ login: "
 
 /* On SVR4, maybe on other systems too, you can cause the 'login' program
  * to prompt with the same string as mgetty did, instead of the standard
@@ -221,7 +221,7 @@
  *
  * Depending on your system, "/var/run/mgetty.%d" might be a good place.
  */
-#define MGETTY_PID_FILE	"/etc/mg-pid.%s"
+#define MGETTY_PID_FILE	"/var/run/mgetty-pid.%s"
 
 /* Path for the lock files. A %s will be replaced with the device name,
  * e.g. tty2a -> /usr/spool/uucp/LCK..tty2a
@@ -582,14 +582,14 @@
 #endif
 
 #ifndef MAILER
-# define MAILER		"/usr/lib/sendmail"
+# define MAILER		"/usr/sbin/sendmail"
 # define NEED_MAIL_HEADERS
 #endif
 
 /* where to send notify mail about incoming faxes to
  * (remember to create an mail alias if no such user exists!)
  */
-#define MAIL_TO		"faxadmin"
+#define MAIL_TO		"root"
 
 /* after a fax has arrived, mgetty can call a program for further
  * processing of this fax.
@@ -604,7 +604,7 @@
  * If you don't want this type of service, do not define it at all
  * Absolute path name has to be used here!
  */
-#define FAX_NOTIFY_PROGRAM "/usr/local/lib/mgetty+sendfax/new_fax"
+#define FAX_NOTIFY_PROGRAM "/etc/mgetty+sendfax/new_fax"
 
 /* default minimum space required on spooling partition for receiving a FAX
  * (in KILObytes)
