--- /dev/null	Tue May  5 22:32:27 1998
+++ mgetty-1.1.14/logrotate.mgetty	Sat Aug 22 23:46:04 1998
@@ -0,0 +1,9 @@
+#
+# Sample entry for rotating mgetty log files.
+# One entry should be present for each device using mgetty.
+#
+
+#/var/log/mgetty.log.ttyS0 {
+#	nocompress
+#	missingok
+#}
--- /dev/null	Tue May  5 22:32:27 1998
+++ mgetty-1.1.14/logrotate.sendfax	Sat Aug 22 23:46:04 1998
@@ -0,0 +1,4 @@
+/var/log/sendfax.log {
+	nocompress
+	missingok	
+}
