--- mgetty-1.1.9/frontends/X11/viewfax-2.4/Imakefile.ORIGINAL	Sun Sep 21 14:48:34 1997
+++ mgetty-1.1.9/frontends/X11/viewfax-2.4/Imakefile	Sun Sep 21 14:49:55 1997
@@ -5,9 +5,9 @@
    complain about this.  Sheesh! :-) */
 HELPDIR = ${USRLIBDIR}/X11
 #else
-BINDIR = /usr/local/bin		/* where you want the binary installed */
-MANDIR = /usr/local/man/man1	/* where you want the man file installed */
-HELPDIR = /usr/local/lib	/* where you want the help file installed */
+BINDIR = /usr/bin		/* where you want the binary installed */
+MANDIR = /usr/man/man1	/* where you want the man file installed */
+HELPDIR = /usr/lib/mgetty+sendfax	/* where you want the help file installed */
 #endif
 
 DEFINES = -DHELPFILE=\"$(HELPDIR)/viewfax.tif\"
