--- file-3.26/Magdir/elf.sparcv9	Sat Sep 12 09:07:39 1998
+++ file-3.26/Magdir/elf	Thu Dec 17 16:13:30 1998
@@ -101,7 +101,7 @@
 >>18	beshort		40		Advanced RISC Machines ARM,
 >>18	beshort		41		Alpha,
 >>18	leshort		42		Hitachi SH,
->>18	beshort		43		SPARC V9 - invalid byte order,
+>>18	beshort		43		SPARC V9,
 >>18	beshort		44		Siemens Tricore Embedded Processor,
 >>18	beshort		45		Argonaut RISC Core, Argonaut Technologies Inc.,
 >>18	beshort		46		Hitachi H8/300,
