--- linux-2.2.5/arch/alpha/kernel/setup.c.msw	Sun Mar 31 06:17:35 2019
+++ linux-2.2.5/arch/alpha/kernel/setup.c	Sun Mar 31 06:17:56 2019
@@ -716,7 +716,8 @@
 		      "BogoMIPS\t\t: %lu.%02lu\n"
 		      "kernel unaligned acc\t: %ld (pc=%lx,va=%lx)\n"
 		      "user unaligned acc\t: %ld (pc=%lx,va=%lx)\n"
-		      "platform string\t\t: %s\n",
+		      "platform string\t\t: %s\n"
+		      "cpus detected\t\t: %ld\n",
 		       cpu_name, cpu->variation, cpu->revision,
 		       (char*)cpu->serial_no,
 		       systype_name, sysvariation_name, hwrpb->sys_revision,
@@ -731,7 +732,7 @@
 		       loops_per_sec / 500000, (loops_per_sec / 5000) % 100,
 		       unaligned[0].count, unaligned[0].pc, unaligned[0].va,
 		       unaligned[1].count, unaligned[1].pc, unaligned[1].va,
-		       platform_string());
+		       platform_string(), hwrpb->nr_processors);
 
 #ifdef __SMP__
 	len += smp_info(buffer+len);
