--- xpat2-1.04/src/Imakefile.nobr	Thu Sep 17 22:55:32 1998
+++ xpat2-1.04/src/Imakefile	Thu Sep 17 23:04:17 1998
@@ -166,48 +166,48 @@
 all::	$(MYPROG)
 
 install:: $(MYPROGS)
-	mkdirhier $(XPATLIBDIR)
-	chmod 755 $(XPATLIBDIR)
+	mkdir -p $(DESTDIR)$(XPATLIBDIR)
+	chmod 755 $(DESTDIR)$(XPATLIBDIR)
 	for f in $(LANGUAGES); do \
-	  mkdirhier $(APPDEFSDIR)/$$f/app-defaults; \
-	  cp ../lib/$$f/XPat $(APPDEFSDIR)/$$f/app-defaults; \
-	  chmod 755 $(APPDEFSDIR)/$$f/app-defaults; \
-	  chmod 644 $(APPDEFSDIR)/$$f/app-defaults/XPat; \
+	  mkdir -p $(DESTDIR)$(APPDEFSDIR)/$$f/app-defaults; \
+	  cp ../lib/$$f/XPat $(DESTDIR)$(APPDEFSDIR)/$$f/app-defaults; \
+	  chmod 755 $(DESTDIR)$(APPDEFSDIR)/$$f/app-defaults; \
+	  chmod 644 $(DESTDIR)$(APPDEFSDIR)/$$f/app-defaults/XPat; \
 	done 
-	(cd ../lib; tar cf - * | (cd $(XPATLIBDIR); tar xf -))
-	chown -R root $(XPATLIBDIR)
-	chmod -R a+r $(XPATLIBDIR)
+	(cd ../lib; tar cf - * | (cd $(DESTDIR)$(XPATLIBDIR); tar xf -))
+	-chown -R root $(DESTDIR)$(XPATLIBDIR)
+	chmod -R a+r $(DESTDIR)$(XPATLIBDIR)
 	for f in $(LANGUAGES); do \
-	    chmod 755 $(XPATLIBDIR)/$$f; \
+	    chmod 755 $(DESTDIR)$(XPATLIBDIR)/$$f; \
 	done
-	mkdirhier $(XPATMANDIR)
-	cp xpat2.man $(XPATMANDIR)/xpat2.6x
-	chmod 644 $(XPATMANDIR)/xpat2.6x
+	mkdir -p $(DESTDIR)$(XPATMANDIR)
+	cp xpat2.man $(DESTDIR)$(XPATMANDIR)/xpat2.6x
+	chmod 644 $(DESTDIR)$(XPATMANDIR)/xpat2.6x
 
 ComplexProgramTarget($(MYPROG))
 InstallAppDefaults(XPat)
 
 install.fsstnd:
 	$(MAKE) install
-	mkdirhier /usr/games/bin
+	mkdir -p /usr/games/bin
 	chmod 755 /usr/games/bin
 	mv $(BINDIR)/xpat2 /usr/games/bin
 
 install.local: $(MYPROGS)
-	mkdirhier $(LXPATBINDIR) $(LXPATMANDIR) $(XPATLIBDIR) $(LAPPDEFSDIR)/app-defaults
-	chmod 755 $(XPATLIBDIR) $(LXPATMANDIR) $(LAPPDEFSDIR)/app-defaults
+	mkdir -p $(LXPATBINDIR) $(LXPATMANDIR) $(DESTDIR)$(XPATLIBDIR) $(LAPPDEFSDIR)/app-defaults
+	chmod 755 $(DESTDIR)$(XPATLIBDIR) $(LXPATMANDIR) $(LAPPDEFSDIR)/app-defaults
 	for f in $(LANGUAGES); do \
-	  mkdirhier $(LAPPDEFSDIR)/$$f/app-defaults; \
+	  mkdir -p $(LAPPDEFSDIR)/$$f/app-defaults; \
 	  cp ../lib/$$f/XPat $(LAPPDEFSDIR)/$$f/app-defaults; \
 	  chmod 755 $(LAPPDEFSDIR)/$$f/app-defaults; \
 	  chmod 644 $(LAPPDEFSDIR)/$$f/app-defaults/XPat; \
 	done 
 	cp XPat.ad $(LAPPDEFSDIR)/app-defaults/XPat
 	chmod 755 $(LAPPDEFSDIR)/app-defaults/XPat
-	(cd ../lib; tar cf - * | (cd $(XPATLIBDIR); tar xf -))
-	chmod -R a+r $(XPATLIBDIR)
+	(cd ../lib; tar cf - * | (cd $(DESTDIR)$(XPATLIBDIR); tar xf -))
+	chmod -R a+r $(DESTDIR)$(XPATLIBDIR)
 	for f in $(LANGUAGES); do \
-	    chmod 755 $(XPATLIBDIR)/$$f; \
+	    chmod 755 $(DESTDIR)$(XPATLIBDIR)/$$f; \
 	done
 	cp xpat2 $(LXPATBINDIR)
 	chmod 755 $(LXPATBINDIR)/xpat2
