diff -uNr mc-4.5.33/configure.in mc-4.5.33-pt_BR/configure.in
--- mc-4.5.33/configure.in	Sun Jun 27 23:38:29 1999
+++ mc-4.5.33-pt_BR/configure.in	Sun Jun 27 17:43:00 1999
@@ -66,7 +66,7 @@
     X11_WWW=lynx
 fi
 
-ALL_LINGUAS="cs da de es es_ES fr it ja ko no pl ru ro"
+ALL_LINGUAS="cs da de es es_ES fr it ja ko no pl pt_BR ru ro"
 AM_GNU_GETTEXT
 AC_LINK_FILES($nls_cv_header_libgt, $nls_cv_header_intl)
 
