diff -uNr mkbootdisk-1.1/mkbootdisk mkbootdisk-1.1.acme/mkbootdisk
--- mkbootdisk-1.1/mkbootdisk	Sat Aug 29 13:44:07 1998
+++ mkbootdisk-1.1.acme/mkbootdisk	Wed Dec  9 21:36:36 1998
@@ -108,8 +108,11 @@
 }
 
 [ -n "$pause" ] && {
-    echo "Insert a disk in $device. Any information on the disk will be lost."
-    echo -n "Press <Enter> to continue or ^C to abort: "
+    # echo "Insert a disk in $device. Any information on the disk will be lost."
+    echo "Coloque um disco em $device."
+    echo "Qualquer informacao neste disco será perdida."
+    # echo -n "Press <Enter> to continue or ^C to abort: "
+    echo -n "Pressione <Enter> para continuar ou Control+C para abortar: "
     read aline
 }
 
@@ -168,9 +171,14 @@
 [ -f $MOUNTDIR/initrd.img ] && echo "	initrd=/initrd.img" >> $MOUNTDIR/etc/lilo.conf
 chmod 644 $MOUNTDIR/etc/lilo.conf
 
+#cat >> $MOUNTDIR/boot/message <<EOF
+#Press <return> (or wait 10 seconds) to boot your Red Hat Linux system from
+#$rootdev. If you want to use a rescue disk, type "rescue" now.
+#
+#EOF
 cat >> $MOUNTDIR/boot/message <<EOF
-Press <return> (or wait 10 seconds) to boot your Red Hat Linux system from
-$rootdev. If you want to use a rescue disk, type "rescue" now.
+Pressione <enter> (ou espere 10 segundos) para inicializar seu sistema Conectiva Linux em
+$rootdev. Se desejar usar um disco de recuperação digite "rescue" agora.
 
 EOF
 
