--- netcfg-2.19/Makefile.nobr	Mon Aug 17 13:27:20 1998
+++ netcfg-2.19/Makefile	Mon Aug 17 13:29:04 1998
@@ -27,21 +27,23 @@
 	rm -f *.pyc
 
 install: all
-	mkdir -p $(PANELDIR)
+	mkdir -p $(DESTDIR)$(BINDIR)
 	for i in $(PROGS) ; do \
-		$(INSTALLBIN) $$i $(BINDIR) ; \
+		$(INSTALLBIN) $$i $(DESTDIR)$(BINDIR) ; \
 	done
+	mkdir -p $(DESTDIR)$(PANELDIR)
+	mkdir -p $(DESTDIR)/etc/X11/wmconfig
 	for i in $(PANEL) ; do \
-		$(INSTALLDATA) $$i.init $(PANELDIR) ; \
-		$(INSTALLDATA) $$i.xpm $(PANELDIR) ; \
-		$(INSTALLDATA) $$i.wmconfig /etc/X11/wmconfig/$$i ; \
+		$(INSTALLDATA) $$i.init $(DESTDIR)$(PANELDIR) ; \
+		$(INSTALLDATA) $$i.xpm $(DESTDIR)$(PANELDIR) ; \
+		$(INSTALLDATA) $$i.wmconfig $(DESTDIR)/etc/X11/wmconfig/$$i ; \
 	done
 
-	mkdir -p $(NETCFGDIR)
+	mkdir -p $(DESTDIR)$(NETCFGDIR)
 	for i in $(LIBS) ; do \
-		$(INSTALLBIN) $$i $(NETCFGDIR) ;\
+		$(INSTALLBIN) $$i $(DESTDIR)$(NETCFGDIR) ;\
 		a=`basename $$i .py` ;\
-		$(INSTALLBIN) $$a.pyc $(NETCFGDIR) ;\
+		$(INSTALLBIN) $$a.pyc $(DESTDIR)$(NETCFGDIR) ;\
 	done
 
 clean:
