--- ./termcap.ewt	Thu May  7 11:21:00 1998
+++ ./termcap	Thu May  7 11:23:52 1998
@@ -458,7 +458,10 @@
 # ***************************************************************************
 #
 # From: Eric S. Raymond <esr@snark.thyrsus.com> 9 Nov 1995
-linux|linux console:\
+#
+# added linux-lat as an alias -- theline drawing characters aren't right, but
+# I think everything which uses line-drawing chars uses terminfo anyway -- ewt
+linux|linux-lat|linux console:\
 	:am:eo:mi:ms:ut:xn:xo:\
 	:co#80:it#8:li#25:\
 	:&7=^Z:@7=\E[4~:AL=\E[%dL:DC=\E[%dP:DL=\E[%dM:\
