--- sash-2.1/Makefile.ewt	Tue Aug 25 11:24:24 1998
+++ sash-2.1/Makefile	Tue Aug 25 11:24:33 1998
@@ -2,7 +2,7 @@
 # Makefile for sash
 #
 
-CFLAGS = -O3 -Wall -DHAVE_GZIP
+CFLAGS = $(RPM_OPT_FLAGS) -Wall -DHAVE_GZIP
 LDFLAGS = -static -s
 LIBS = -lz
 
--- sash-2.1/cmds.c.ewt	Tue Aug 25 11:15:20 1998
+++ sash-2.1/cmds.c	Tue Aug 25 11:15:24 1998
@@ -16,7 +16,6 @@
 #include <grp.h>
 #include <utime.h>
 #include <errno.h>
-#include <linux/fs.h>
 
 
 void
