diff -uNr svgalib-1.3.1/Makefile.cfg svgalib-1.3.1.conf/Makefile.cfg
--- svgalib-1.3.1/Makefile.cfg	Mon Dec 21 16:08:58 1998
+++ svgalib-1.3.1.conf/Makefile.cfg	Wed Mar 31 19:40:08 1999
@@ -22,7 +22,7 @@
 
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
-TOPDIR=
+TOPDIR=$(INSTALL_PREFIX)
 prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
@@ -30,7 +30,7 @@
 libdir = $(exec_prefix)/lib
 
 # Directory where the shared library will be installed.
-sharedlibdir = $(TOPDIR)/lib
+sharedlibdir = $(exec_prefix)/lib
 
 # Directory where the font and textmode utilities will be installed.
 bindir = $(exec_prefix)/bin
@@ -51,13 +51,13 @@
 # Uncomment the next line if your are compiling with a version of
 # (GNU) glibc:
 
-# USEGLIBC = yes
+USEGLIBC = yes
 
 # Uncomment this if you want root processes to be able to always get a new
 # VC. Alas, some games misuse suid root privs and become root, svgalib cannot
 # detect this and will allow Joe blow user to open a new virtual VC. If this
 # annoys you, comment out the next line (which is the default already)
-ROOT_VC_SHORTCUT = y
+# ROOT_VC_SHORTCUT = y
 
 # Uncomment to enable run in background mode
 # This doesn't work at all at the moment on linux-alpha-machines
@@ -85,7 +85,7 @@
 ONE_MAP = y
 
 # Uncomment this if you want to compile and install the static libs.
-# INSTALLSTATICLIB = installstaticlib
+INSTALLSTATICLIB = installstaticlib
 
 # Comment this out if you don't want to install the shared libs.
 # If you do not install the shared nor the static libs, 'make static'
@@ -97,7 +97,7 @@
 # a.out image is available install it in the first directory named something
 # like *aout/ in /etc/ld.so.conf.
 # If you want this, do not comment out the next line:
-INSTALLAOUTLIB = installaoutcompat
+# INSTALLAOUTLIB = installaoutcompat
 
 # Comment this out if you want to keep old shared images. Old header files,
 # library stubs and static libraries CANNOT be kept in public locations
@@ -229,7 +229,7 @@
 #----------------------------------------------------------------------
 
 # Compiler used.
-CC       = gcc
+CC       = egcs
 #CC      = gcc -b i486-linuxaout
 #CC      = gcc -b i486-linux
 
diff -uNr svgalib-1.3.1/utils/Makefile svgalib-1.3.1.conf/utils/Makefile
--- svgalib-1.3.1/utils/Makefile	Tue Jul 21 16:42:09 1998
+++ svgalib-1.3.1.conf/utils/Makefile	Wed Mar 31 19:42:14 1999
@@ -17,6 +17,7 @@
 #Use the next one for the Alpha/AXP if you need it
 #LDFLAGS = -L../staticlib 
 LIBS = -lvga -lm
+LIBS := -L$(libdir) $(LIBS)
 
 #----------------------------------------------------------------------
 # Rules Section
