diff -Nur stunnel.orig/common.h stunnel/common.h
--- stunnel.orig/common.h	Thu Jun 10 18:30:58 1999
+++ stunnel/common.h	Thu Jun 10 18:31:38 1999
@@ -25,10 +25,10 @@
 #define RSAref
 #endif
 #define NO_BLOWFISH
-#include <ssl.h>
-#include <err.h>
-#include <lhash.h>
-#include <crypto.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#include <openssl/lhash.h>
+#include <openssl/crypto.h>
 
 #ifdef USE_WIN32
 
