--- sharutils-4.2/doc/Makefile.in~	Fri Nov 24 21:44:35 1995
+++ sharutils-4.2/doc/Makefile.in	Mon Dec  9 10:59:49 1996
@@ -101,7 +101,7 @@
 	for name in $(MAN1PAGES); do \
 	  if test $$name = uudecode; then \
 	    rm -f $(man1dir)/`echo uudecode | sed '$(transform)'`$(man1ext); \
-	    man1base=`echo $(man1dir) | sed 's@.*/\(.*\)@&1@'`; \
+	    man1base=man1; \
 	    echo ".so $$man1base/uuencode$(man1ext)" \
 	      > $(man1dir)/`echo uudecode | sed '$(transform)'`$(man1ext); \
 	  else \
