--- sharutils-4.2/po/Makefile.in.in.marc	Sun Jul 28 19:05:23 1996
+++ sharutils-4.2/po/Makefile.in.in	Sun Jul 28 19:16:44 1996
@@ -124,7 +124,7 @@
 	  else \
 	    destdir=$(localedir); \
 	  fi; \
-	  lang=`echo $$cat | sed 's/$(CATOBJEXT)$$//'`; \
+	  lang=`echo $$cat | sed 's/\.[^.]*$$//'`; \
 	  dir=$$destdir/$$lang/LC_MESSAGES; \
 	  $(top_srcdir)/mkinstalldirs $$dir; \
 	  if test -r $$cat; then \
