--- tcsh-6.07.09/sh.h.termios	Tue Jul  7 08:06:20 1998
+++ tcsh-6.07.09/sh.h	Fri Oct  2 15:00:53 1998
@@ -194,6 +194,8 @@
 # define DEV_DEV_COMPARE(x,y)   ((x) == (y))
 #endif /* DEV_DEV_COMPARE */
 
+#define POSIX
+
 #ifdef _SEQUENT_
 # include <sys/procstats.h>
 #endif /* _SEQUENT_ */
@@ -235,12 +237,10 @@
 #  endif /* TERMIO */
 # else /* POSIX */
 #  include <termios.h>
-#  if SYSVREL > 3
 #   undef TIOCGLTC	/* we don't need those, since POSIX has them */
 #   undef TIOCSLTC
 #   undef CSWTCH
 #   define CSWTCH _POSIX_VDISABLE	/* So job control works */
-#  endif /* SYSVREL > 3 */
 # endif /* POSIX */
 #endif /* WINNT */
 
@@ -298,6 +298,8 @@
 #if !((defined(SUNOS4) || defined(_MINIX) /* || defined(DECOSF1) */) && defined(TERMIO))
 # if !defined(COHERENT) && !defined(_VMS_POSIX) && !defined(WINNT)
 #  include <sys/ioctl.h>
+#  undef TIOCGLTC	/* we don't need those, since POSIX has them */
+#  undef TIOCSLTC
 # endif
 #endif 
 
