--- xc/programs/Xserver/hw/xfree86/XF86Conf.man.org	Mon Jun 28 05:09:01 1999
+++ xc/programs/Xserver/hw/xfree86/XF86Conf.man	Tue Jun 29 03:17:58 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86Conf.man,v 3.52.2.12 1999/06/02 07:50:28 hohndel Exp $
-.TH XF86Config 4/5 "Version 3.3.3"  "XFree86"
+.TH XF86Config 4/5 "Version 3.3.4"  "XFree86"
 .SH NAME
 XF86Config - Configuration File for XFree86
 .SH DESCRIPTION
--- xc/programs/Xserver/hw/xfree86/XFree86.man.org	Mon Jun 28 05:09:01 1999
+++ xc/programs/Xserver/hw/xfree86/XFree86.man	Tue Jun 29 03:36:25 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XFree86.man,v 3.36.2.11 1999/06/02 07:50:29 hohndel Exp $ 
-.TH XFree86 1 "Version 3.3.3"  "XFree86"
+.TH XFree86 1 "Version 3.3.4"  "XFree86"
 .SH NAME
 XFree86 - X11R6 for UNIX on x86 platforms
 .SH DESCRIPTION
@@ -22,11 +22,12 @@
 .br
 -- Solaris (x86) 2.1, 2.4, 2.5, 2.5.1, 2.6
 .br
--- FreeBSD 2.0.5, 2.1, 2.1.5, 2.1.6, 2.1.7, 2.1.7.1, 2.2, 2.2.1, 2.2.2, 2.2.5, 2.2.6, 3.0-current
+-- FreeBSD 2.0.5, 2.1, 2.1.5, 2.1.6, 2.1.7, 2.1.7.1, 2.2, 2.2.1,
+2.2.2, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 3.0, 3.1, 3.2
 .br
--- NetBSD 1.0, 1.1, 1.2, 1.2.1, 1.3 (i386 port only)
+-- NetBSD 1.0, 1.1, 1.2, 1.2.1, 1.3, 1.3.1, 1.3.2, 1.3.3 (i386 port only)
 .br
--- OpenBSD 2.0, 2.1
+-- OpenBSD 2.0, 2.1, 2.2, 2.3, 2.4
 .br
 -- BSD/386 version 1.1 and BSD/OS 2.0
 .br
--- xc/programs/Xserver/hw/xfree86/XF86_Acc.man.org	Sun Nov  8 09:01:00 1998
+++ xc/programs/Xserver/hw/xfree86/XF86_Acc.man	Tue Jun 29 03:18:58 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86_Acc.man,v 3.58.2.9 1998/11/06 09:45:57 hohndel Exp $ 
-.TH XF86_Accel 1 "Version 3.3.3"  "XFree86"
+.TH XF86_Accel 1 "Version 3.3.4"  "XFree86"
 .SH NAME
 XF86_Accel - accelerated X Window System servers for
 UNIX on x86 and Alpha platforms with an S3, Mach8, Mach32, Mach64, P9000, AGX,
--- xc/programs/Xserver/hw/xfree86/XF86_Mono.man.org	Sun Nov  8 09:01:01 1998
+++ xc/programs/Xserver/hw/xfree86/XF86_Mono.man	Tue Jun 29 03:19:28 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86_Mono.man,v 3.23.2.5 1998/11/06 09:45:59 hohndel Exp $
-.TH XF86_Mono 1 "Version 3.3.3"  "XFree86"
+.TH XF86_Mono 1 "Version 3.3.4"  "XFree86"
 .SH NAME
 XF86_Mono - 1 bit non-accelerated X Window System servers for UNIX on
 x86 platforms
--- xc/programs/Xserver/hw/xfree86/XF86_SVGA.man.org	Mon Jun 28 05:02:12 1999
+++ xc/programs/Xserver/hw/xfree86/XF86_SVGA.man	Tue Jun 29 03:32:22 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86_SVGA.man,v 3.42.2.10 1999/04/15 12:04:24 hohndel Exp $ 
-.TH XF86_SVGA 1 "Version 3.3.3"  "XFree86"
+.TH XF86_SVGA 1 "Version 3.3.4"  "XFree86"
 .SH NAME
 XF86_SVGA - SVGA X Window System servers for UNIX on 
 x86 platforms
@@ -53,7 +53,10 @@
 Trident:
 TVGA8800CS, TVGA8900B, TVGA8900C, TVGA8900CL, TVGA9000
 TVGA9000i, TVGA9100B, TVGA9200CXr, TGUI9320LCD, TGUI9400CXi, TGUI9420,
-TGUI9420DGi, TGUI9430DGi, TGUI9440AGi, TGUI9660XGi, TGUI9680
+TGUI9420DGi, TGUI9430DGi, TGUI9440AGi, TGUI9660XGi, TGUI9680,
+PROVIDIA9682, PROVIDIA9685, CYBER9382, CYBER9385, CYBER9385-1,
+CYBER9388, CYBER9397, CYBER9520, CYBER9525, 3DIMAGE975, 3DIMAGE975 AGP,
+3DIMAGE985 AGP, BLADE3D, CYBERBLADE
 .TP 4
 NCR:
 77C22, 77C22E
@@ -99,7 +102,8 @@
 AP6422
 .TP 4
 Matrox:
-MGA2064W
+MGA1064SG, MGA1164SG, MGA2064W, MGA2164W, MGA2164 AGP,
+MGAG100, MGAG200, MGAG400
 .TP 4
 NVidia:
 NV1, STG2000
--- xc/programs/Xserver/hw/xfree86/XF86_VGA16.man.org	Sun Nov  8 09:01:01 1998
+++ xc/programs/Xserver/hw/xfree86/XF86_VGA16.man	Tue Jun 29 03:32:48 1999
@@ -1,5 +1,5 @@
 .\" $XFree86: xc/programs/Xserver/hw/xfree86/XF86_VGA16.man,v 3.22.2.5 1998/11/06 09:46:02 hohndel Exp $
-.TH XF86_VGA16 1 "Version 3.3.3"  "XFree86"
+.TH XF86_VGA16 1 "Version 3.3.4"  "XFree86"
 .SH NAME
 XF86_VGA16 - 4 bit non-accelerated X Window System server for UNIX on
 x86 platforms
--- xc/programs/Xserver/hw/xfree86/doc/man/XF86DGA.man.org	Sun Nov  8 09:01:05 1998
+++ xc/programs/Xserver/hw/xfree86/doc/man/XF86DGA.man	Tue Jun 29 03:37:05 1999
@@ -5,7 +5,7 @@
 .ie t \fB\^\\$1\^\fR\\$2
 .el \fI\^\\$1\^\fP\\$2
 ..
-.TH XF86DGA 3X11 "3.3.3" "XFree86" "X FUNCTIONS"
+.TH XF86DGA 3X11 "3.3.4" "XFree86" "X FUNCTIONS"
 .SH NAME
 XF86DGAQueryExtension, XF86DGAQueryVersion, XF86DGAQueryDirectVideo, XF86DGAGetVideo, XF86DGADirectVideo, XF86DGASetVidPage, XF86DGASetViewPort, XF86DGAViewPortChanged, XF86DGAGetViewPortSize, XF86DGAInstallColormap, XF86DGAForkApp \- XFree86-DGA extension interface functions
 .SH SYNTAX
--- xc/programs/Xserver/hw/xfree86/doc/man/XF86Misc.man.org	Sun Nov  8 09:01:05 1998
+++ xc/programs/Xserver/hw/xfree86/doc/man/XF86Misc.man	Tue Jun 29 03:37:43 1999
@@ -9,7 +9,7 @@
 .ie t \fB\^\\$1\^\fR\\$2
 .el \fI\^\\$1\^\fP\\$2
 ..
-.TH XF86MISC 3X11 "3.3.3" "XFree86" "X FUNCTIONS"
+.TH XF86MISC 3X11 "3.3.4" "XFree86" "X FUNCTIONS"
 .SH NAME
 XF86MiscQueryExtension, XF86MiscQueryVersion, XF86MiscGetMouseSettings, XF86MiscSetMouseSettings, XF86MiscGetKbdSettings, XF86MiscSetKbdSettings \- XFree86-Misc extension interface functions
 .SH SYNTAX
--- xc/programs/Xserver/hw/xfree86/doc/man/XF86VM.man.org	Sun Nov  8 09:01:05 1998
+++ xc/programs/Xserver/hw/xfree86/doc/man/XF86VM.man	Tue Jun 29 03:38:10 1999
@@ -10,7 +10,7 @@
 .ie t \fB\^\\$1\^\fR\\$2
 .el \fI\^\\$1\^\fP\\$2
 ..
-.TH XF86VIDMODE 3X11 "3.3.3" "XFree86" "X FUNCTIONS"
+.TH XF86VIDMODE 3X11 "3.3.4" "XFree86" "X FUNCTIONS"
 .SH NAME
 XF86VidModeQueryExtension, XF86VidModeQueryVersion, XF86VidModeGetModeLine, XF86VidModeGetAllModeLines, XF86VidModeDeleteModeLine, XF86VidModeModModeLine, XF86VidModeValidateModeLine, XF86VidModeSwitchMode, XF86VidModeSwitchToMode, XF86VidModeLockModeSwitch, XF86VidModeGetMonitor, XF86VidModeGetViewPort, XF86VidModeSetViewPort \- XFree86-VidMode extension interface functions
 .SH SYNTAX
