--- XFree86-3.3.3.1/xc/programs/xinit/startx.cpp.startx_xauth	Tue Feb  2 18:44:23 1999
+++ XFree86-3.3.3.1/xc/programs/xinit/startx.cpp	Tue Feb  2 18:51:48 1999
@@ -1,75 +1,22 @@
 XCOMM!/bin/sh
-
-XCOMM $XConsortium: startx.cpp,v 1.4 91/08/22 11:41:29 rws Exp $
-XCOMM $XFree86: xc/programs/xinit/startx.cpp,v 3.0.8.4 1998/12/27 13:10:18 dawes Exp $
-XCOMM 
-XCOMM This is just a sample implementation of a slightly less primitive 
-XCOMM interface than xinit.  It looks for user .xinitrc and .xserverrc
-XCOMM files, then system xinitrc and xserverrc files, else lets xinit choose
-XCOMM its default.  The system xinitrc should probably do things like check
-XCOMM for .Xresources files and merge them in, startup up a window manager,
-XCOMM and pop a clock and serveral xterms.
 XCOMM
-XCOMM Site administrators are STRONGLY urged to write nicer versions.
-XCOMM 
+XCOMM (c) 1999 Red Hat Software, Inc.
 
 bindir=BINDIR
-#ifdef SCO
-
-XCOMM Check for /usr/bin/X11 and BINDIR in the path, if not add them.
-XCOMM This allows startx to be placed in a place like /usr/bin or /usr/local/bin
-XCOMM and people may use X without changing their PATH
-
-XCOMM First our compiled path
-
-if expr $PATH : ".*`echo $bindir | sed 's?/?\\/?g'`.*" > /dev/null 2>&1; then
-	:
-else
-	PATH=$PATH:BINDIR
-fi
-
-XCOMM Now the "SCO" compiled path
-
-if expr $PATH : '.*\/usr\/bin\/X11.*' > /dev/null 2>&1; then
-	:
-else
-	PATH=$PATH:/usr/bin/X11
-fi
-
-XCOMM Set up the XMERGE env var so that dos merge is happy under X
-
-if [ -f /usr/lib/merge/xmergeset.sh ]; then
-	. /usr/lib/merge/xmergeset.sh
-else if [ -f /usr/lib/merge/console.disp ]; then
-	XMERGE=`cat /usr/lib/merge/console.disp`
-	export XMERGE
-fi
-fi
-
-scoclientrc=$HOME/.startxrc
-#endif
 
 userclientrc=$HOME/.xinitrc
 userserverrc=$HOME/.xserverrc
-sysclientrc=XINITDIR/xinitrc
-sysserverrc=XINITDIR/xserverrc
+sysclientrc=/etc/X11/xinit/xinitrc
+sysserverrc=/etc/X11/xinit/xserverrc
 clientargs=""
 serverargs=""
 
-#ifdef SCO
-if [ -f $scoclientrc ]; then
-    clientargs=$scoclientrc
-else
-#endif
 if [ -f $userclientrc ]; then
     clientargs=$userclientrc
 else if [ -f $sysclientrc ]; then
     clientargs=$sysclientrc
 fi
 fi
-#ifdef SCO
-fi
-#endif
 
 if [ -f $userserverrc ]; then
     serverargs=$userserverrc
@@ -78,6 +25,7 @@
 fi
 fi
 
+display=:0
 whoseargs="client"
 while [ "x$1" != "x" ]; do
     case "$1" in
@@ -99,16 +47,26 @@
 		    clientargs="$clientargs $1"
 		else
 		    serverargs="$serverargs $1"
+		case "$1" in
+		    :[0-9]) display="$1"
+		    ;;
+		esac
 		fi ;;
     esac
     shift
 done
 
+XCOMM set up default Xauth info for this machine
+mcookie=`mcookie`
+serverargs="$serverargs -auth $HOME/.Xauthority"
+xauth add $display . $mcookie
+xauth add `hostname -f`$display . $mcookie
+
 xinit $clientargs -- $serverargs
 
-/*
- * various machines need special cleaning up
- */
+XCOMM various machines need special cleaning up,
+XCOMM which should be done here
+
 #ifdef macII
 Xrepair
 screenrestore
