--- gdm-1.0.0/src/gdmgreeter.c.nosuspend	Fri Apr  9 16:26:07 1999
+++ gdm-1.0.0/src/gdmgreeter.c	Fri Apr  9 16:26:34 1999
@@ -1072,6 +1072,7 @@
 			   GTK_SIGNAL_FUNC (gdm_greeter_halt_handler), NULL);
 	gtk_widget_show(item);
 
+#if 0
 	if(GdmSuspend) {
 	    item = gtk_menu_item_new_with_label (_("Suspend"));
 	    gtk_menu_append(GTK_MENU (submenu), item);
@@ -1079,6 +1080,7 @@
 			       GTK_SIGNAL_FUNC (gdm_greeter_suspend_handler), NULL);
 	    gtk_widget_show(item);
 	}
+#endif
 
 	item = gtk_menu_item_new_with_label (_("System"));
 	gtk_menu_append(GTK_MENU (optionmenu), item);
