--- tcltk-8.0.3/expect-5.26/exp_strf.c.glibc21	Tue Feb 16 13:31:45 1999
+++ tcltk-8.0.3/expect-5.26/exp_strf.c	Tue Feb 16 13:32:05 1999
@@ -98,7 +98,6 @@
 adddecl(static int iso8601wknum();)
 #else
 
-extern char *strchr(const char *str, int ch);
 extern char *getenv(const char *v);
 static int weeknumber(const struct tm *timeptr, int firstweekday);
 adddecl(static int iso8601wknum(const struct tm *timeptr);)
