--- gzip-1.2.4/gzip.h.orig	Wed Nov  1 23:02:14 1995
+++ gzip-1.2.4/gzip.h	Wed Nov  1 23:02:20 1995
@@ -301,7 +301,7 @@
 extern void flush_window  OF((void));
 extern void write_buf     OF((int fd, voidp buf, unsigned cnt));
 extern char *strlwr       OF((char *s));
-extern char *basename     OF((char *fname));
+extern char *basename     OF((const char *fname));
 extern void make_simple_name OF((char *name));
 extern char *add_envopt   OF((int *argcp, char ***argvp, char *env));
 extern void error         OF((char *m));
--- gzip-1.2.4/util.c.orig	Wed Nov  1 23:03:32 1995
+++ gzip-1.2.4/util.c	Wed Nov  1 23:03:37 1995
@@ -179,7 +179,7 @@
  * case sensitive, force the base name to lower case.
  */
 char *basename(fname)
-    char *fname;
+    const char *fname;
 {
     char *p;
 
