--- shadow-980403/libmisc/Makefile.am.nscd	Sat Dec 20 14:36:56 1997
+++ shadow-980403/libmisc/Makefile.am	Thu Apr  8 17:08:11 1999
@@ -13,7 +13,7 @@
  chowndir.c chowntty.c console.c copydir.c entry.c env.c failure.c \
  fields.c getdate.y hushed.c isexpired.c limits.c list.c log.c \
  login_access.c login_desrpc.c login_krb.c loginprompt.c mail.c motd.c \
- myname.c obscure.c pam_pass.c pwd2spwd.c pwd_init.c rlogin.c salt.c \
+ myname.c nscd.c obscure.c pam_pass.c pwd2spwd.c pwd_init.c rlogin.c salt.c \
  setugid.c setup.c setupenv.c shell.c strtoday.c suauth.c sub.c sulog.c \
  ttytype.c tz.c ulimit.c utmp.c valid.c xmalloc.c
 
--- shadow-980403/libmisc/nscd.c.nscd	Thu Apr  8 17:08:58 1999
+++ shadow-980403/libmisc/nscd.c	Thu Apr  8 17:08:11 1999
@@ -0,0 +1,27 @@
+/* club nscd so it invalidates its cache */
+/* notting@redhat.com 19990408           */
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+
+int hup_nscd() {
+    FILE *pidfile;
+    char *pidbuf=calloc(256,sizeof(char));
+    int pid;
+    
+    if ((pidfile=fopen("/var/run/nscd.pid","r"))!=NULL) {
+	pidbuf=fgets(pidbuf,256,pidfile);
+	fclose(pidfile);
+	if (pidbuf!=NULL) {
+	    pid=atoi(pidbuf);
+	    free(pidbuf);
+	    if (pid>0) {
+		return kill(pid,SIGHUP);
+	    }
+	}
+    }
+    return -1;
+}
--- shadow-980403/libmisc/Makefile.in.nscd	Thu Apr  2 16:57:57 1998
+++ shadow-980403/libmisc/Makefile.in	Thu Apr  8 17:08:42 1999
@@ -81,7 +81,7 @@
  chowndir.c chowntty.c console.c copydir.c entry.c env.c failure.c \
  fields.c getdate.y hushed.c isexpired.c limits.c list.c log.c \
  login_access.c login_desrpc.c login_krb.c loginprompt.c mail.c motd.c \
- myname.c obscure.c pam_pass.c pwd2spwd.c pwd_init.c rlogin.c salt.c \
+ myname.c nscd.c obscure.c pam_pass.c pwd2spwd.c pwd_init.c rlogin.c salt.c \
  setugid.c setup.c setupenv.c shell.c strtoday.c suauth.c sub.c sulog.c \
  ttytype.c tz.c ulimit.c utmp.c valid.c xmalloc.c
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
--- shadow-980403/src/userdel.c.nscd	Thu Apr  8 17:08:11 1999
+++ shadow-980403/src/userdel.c	Thu Apr  8 17:08:11 1999
@@ -857,6 +857,7 @@
 	user_cancel(user_name);
 
 	close_files ();
+        hup_nscd ();
 
 	exit(errors ? E_HOMEDIR : E_SUCCESS);
 	/*NOTREACHED*/
--- shadow-980403/src/useradd.c.nscd	Thu Apr  8 17:08:11 1999
+++ shadow-980403/src/useradd.c	Thu Apr  8 17:08:11 1999
@@ -1921,6 +1921,7 @@
 	}
 
 	close_files ();
+        hup_nscd ();
 
 	exit(E_SUCCESS);
 	/*NOTREACHED*/
--- shadow-980403/src/newusers.c.nscd	Thu Jan 29 18:22:43 1998
+++ shadow-980403/src/newusers.c	Thu Apr  8 17:08:11 1999
@@ -561,6 +561,7 @@
 		spw_unlock();
 #endif
 	(void) pw_unlock ();
+        hup_nscd ();
 
 	exit (0);
 	/*NOTREACHED*/
--- shadow-980403/src/groupmod.c.nscd	Thu Apr  2 16:51:54 1998
+++ shadow-980403/src/groupmod.c	Thu Apr  8 17:08:11 1999
@@ -533,6 +533,7 @@
 	grp_update ();
 
 	close_files ();
+        hup_nscd ();
 	exit(E_SUCCESS);
 	/*NOTREACHED*/
 }
--- shadow-980403/src/groupdel.c.nscd	Thu Jan 29 18:22:40 1998
+++ shadow-980403/src/groupdel.c	Thu Apr  8 17:08:11 1999
@@ -342,6 +342,7 @@
 	grp_update ();
 
 	close_files ();
+        hup_nscd ();
 	exit(errors == 0 ? E_SUCCESS : E_GRP_UPDATE);
 	/*NOTREACHED*/
 }
--- shadow-980403/src/groupadd.c.nscd	Thu Apr  8 17:08:11 1999
+++ shadow-980403/src/groupadd.c	Thu Apr  8 17:08:11 1999
@@ -544,6 +544,7 @@
 	grp_update();
 
 	close_files();
+        hup_nscd();
 	exit(E_SUCCESS);
 	/*NOTREACHED*/
 }
--- shadow-980403/src/gpasswd.c.nscd	Thu Jan 29 18:22:40 1998
+++ shadow-980403/src/gpasswd.c	Thu Apr  8 17:08:11 1999
@@ -652,6 +652,7 @@
 	endsgent ();
 #endif
 #endif
+        hup_nscd ();
 	exit (0);
 	/*NOTREACHED*/
 }
--- shadow-980403/src/usermod.c.nscd	Thu Apr 15 19:55:40 1999
+++ shadow-980403/src/usermod.c	Thu Apr 15 19:56:00 1999
@@ -1681,6 +1681,7 @@
 			user_id, user_newid,
 			user_gid, gflg ? user_newgid:user_gid);
 	}
+        hup_nscd();
 
 	if (grp_err)
 		exit(E_GRP_UPDATE);
