diff -ru ispell-3.1/config.X ispell-3.1.new/config.X
--- ispell-3.1/config.X	Sat Jun 27 20:01:14 1998
+++ ispell-3.1.new/config.X	Sat Jun 27 18:47:32 1998
@@ -511,7 +511,7 @@
 ** information.
 */
 #ifndef MASKBITS
-#define MASKBITS	32
+#define MASKBITS       64
 #endif
 
 /*
@@ -530,7 +530,7 @@
 #define MASKTYPE	long
 #endif
 #ifndef MASKTYPE_WIDTH
-#define MASKTYPE_WIDTH	32
+#define MASKTYPE_WIDTH 64
 #endif
 #if MASKBITS < MASKTYPE_WIDTH
 #undef MASKBITS
