--- ispell-3.1/ijoin.c.strcmp	Tue Jan 12 20:48:11 1999
+++ ispell-3.1/ijoin.c	Tue Jan 12 20:49:02 1999
@@ -89,6 +89,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 #include "config.h"
 #include "ispell.h"
 #include "proto.h"
@@ -133,7 +134,6 @@
 static int		unpairable1 = 0; /* NZ if -a1 */
 static int		unpairable2 = 0; /* NZ if -a2 */
 
-extern int	strcmp ();
 
 int main (argc, argv)			/* Join files */
     int			argc;		/* Argument count */
