--- gnome-core-1.0.3/panel/panel.c.kdemenus	Mon Mar 15 00:05:09 1999
+++ gnome-core-1.0.3/panel/panel.c	Mon Mar 15 00:05:19 1999
@@ -943,7 +943,10 @@
 			  /*guess redhat menus*/
 			  if(g_file_exists("/etc/X11/wmconfig"))
 				  flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
-		    load_menu_applet(ltmp->data, flags, panel, pos);
+			  /* Guess KDE menus */
+			  if(g_file_exists("/usr/share/applnk"))
+				  flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
+			  load_menu_applet(ltmp->data, flags, panel, pos);
 		  } else if(S_IEXEC & s.st_mode) /*executable?*/
 		    ask_about_launcher(ltmp->data,panel,pos);
 		}
@@ -988,6 +991,9 @@
 		/*guess redhat menus*/
 		if(g_file_exists("/etc/X11/wmconfig"))
 			flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
+		/* Guess KDE menus */
+		if(g_file_exists("/usr/share/applnk"))
+				  flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
 		load_menu_applet ((char *)selection_data->data,flags,
 				  panel, pos);
 		break;
@@ -1011,6 +1017,9 @@
 			/*guess redhat menus*/
 			if(g_file_exists("/etc/X11/wmconfig"))
 				flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
+			/* Guess KDE menus */
+			if(g_file_exists("/usr/share/applnk"))
+				flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
 
 			load_menu_applet(NULL,flags, panel, pos);
 		} else if(strcmp(applet_type,"DRAWER:NEW")==0) {
--- gnome-core-1.0.3/panel/session.c.kdemenus	Mon Mar 15 00:05:16 1999
+++ gnome-core-1.0.3/panel/session.c	Mon Mar 15 00:05:19 1999
@@ -738,6 +738,9 @@
 	/*guess redhat menus*/
 	if(g_file_exists("/etc/X11/wmconfig"))
 		flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
+	/*guess KDE menus */
+	if(g_file_exists("/usr/share/applnk"))
+		flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
 	/*guess debian menus*/
 	if (g_file_exists("/etc/menu-methods/gnome"))
 		flags |= MAIN_MENU_DEBIAN|MAIN_MENU_DEBIAN_SUB;
@@ -854,6 +857,9 @@
 				/*guess redhat menus*/
 				if(g_file_exists("/etc/X11/wmconfig"))
 					flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
+				/*guess KDE menus */
+				if(g_file_exists("/usr/share/applnk"))
+					flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
 				/*guess debian menus*/
 				if (g_file_exists("/etc/menu-methods/gnome"))
 					flags |= MAIN_MENU_DEBIAN|MAIN_MENU_DEBIAN_SUB;
--- gnome-core-1.0.3/panel/menu.c.kdemenus	Mon Mar 15 00:05:36 1999
+++ gnome-core-1.0.3/panel/menu.c	Mon Mar 15 00:10:16 1999
@@ -465,7 +465,9 @@
 		} else {
 			GnomeDesktopEntry *dentry;
 			char *p = strrchr(name,'.');
-			if(p && strcmp(p,".desktop")!=0)
+			if (!p || (strcmp(p,".desktop") != 0 &&
+				   strcmp(p,".kdelnk") != 0))
+				  
 				goto next;
 
 			dentry = gnome_desktop_entry_load(name);
@@ -867,7 +869,9 @@
 						     newpanel);
 			} else {
 				char *p = strrchr(filename,'.');
-				if(p && strcmp(p,".desktop")==0)
+				if (p &&
+				    (strcmp(p,".desktop")==0 || 
+				     strcmp(p,".kdelnk")==0))
 					/*we load the applet at the right
 					  side, that is end of the drawer*/
 					load_launcher_applet(filename,
@@ -900,6 +904,8 @@
 	/*guess redhat menus*/
 	if(g_file_exists("/etc/X11/wmconfig"))
 		flags |= MAIN_MENU_REDHAT|MAIN_MENU_REDHAT_SUB;
+	if(g_file_exists("/usr/share/applnk"))
+		flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
 
 	/*guess debian menus*/
 	if (g_file_exists("/etc/menu-methods/gnome"))
@@ -983,6 +989,13 @@
 	
 	g_return_if_fail(item_loc!=NULL);
 
+	dentry = gnome_desktop_entry_load(item_loc);
+	/* We'll screw up a KDE menu entry if we edit it */
+	if (dentry && dentry->is_kde) {
+		gnome_desktop_entry_free (dentry);
+		return;
+	}
+
 	dialog = gnome_property_box_new();
 	gtk_window_set_wmclass(GTK_WINDOW(dialog),
 			       "desktop_entry_properties","Panel");
@@ -993,7 +1006,6 @@
 	/*item loc will be alive all this time*/
 	gtk_object_set_data(o,"location",item_loc);
 
-	dentry = gnome_desktop_entry_load(item_loc);
 	if(dentry) {
 		gnome_dentry_edit_set_dentry(GNOME_DENTRY_EDIT(o),dentry);
 		gnome_desktop_entry_free(dentry);
@@ -1017,6 +1029,13 @@
 	char *dirfile = g_concat_dir_and_file(mf->menudir, ".directory");
 	GnomeDesktopEntry *dentry;
 
+	dentry = gnome_desktop_entry_load_unconditional(dirfile);
+	/* We'll screw up a KDE menu entry if we edit it */
+	if (dentry && dentry->is_kde) {
+		gnome_desktop_entry_free (dentry);
+		return;
+	}
+
 	dialog = gnome_property_box_new();
 	gtk_window_set_wmclass(GTK_WINDOW(dialog),
 			       "desktop_entry_properties","Panel");
@@ -1025,7 +1044,6 @@
 	
 	o = gnome_dentry_edit_new_notebook(GTK_NOTEBOOK(GNOME_PROPERTY_BOX(dialog)->notebook));
 
-	dentry = gnome_desktop_entry_load_unconditional(dirfile);
 	if (dentry) {
 		gnome_dentry_edit_set_dentry(GNOME_DENTRY_EDIT(o), dentry);
 		gtk_object_set_data_full(o,"location",
@@ -1147,7 +1165,7 @@
 					   sim->mf);
 		gtk_object_set_data(GTK_OBJECT(item),"prop_item",
 				    sim->prop_item);
-		setup_menuitem (sim->prop_item, 0, _("Properties ..."));
+		setup_menuitem (sim->prop_item, 0, _("Properties..."));
 		gtk_menu_append (GTK_MENU (sim->menu), sim->prop_item);
 	}
 	
@@ -2323,6 +2341,23 @@
 	return menu;
 }
 
+static GtkWidget *
+create_kde_menu(GtkWidget *menu, int fake_submenus,
+		int force, int fake)
+{
+	if(!fake || menu) {
+		menu = create_menu_at (menu, 
+				       "/usr/share/applnk", FALSE,
+				       _("KDE menus"), 
+				       NULL,fake_submenus,
+				       force);
+	} else {
+		menu = create_fake_menu_at ("/usr/share/applnk", FALSE,
+					    _("KDE menus"),NULL);
+	}
+	return menu;
+}
+
 GtkWidget *
 create_panel_root_menu(GtkWidget *panel)
 {
@@ -3095,6 +3130,10 @@
 		root_menu = create_debian_menu(root_menu, fake_submenus, FALSE);
 		need_separ = TRUE;
 	}
+	if(flags&MAIN_MENU_KDE && !(flags&MAIN_MENU_KDE_SUB)) {
+		root_menu = create_kde_menu(root_menu,fake_submenus,FALSE, FALSE);
+		need_separ = TRUE;
+	}
 	/*others here*/
 	
 	if(!root_menu)
@@ -3171,6 +3210,30 @@
 					   menuitem);
 		}
 	}
+	if(flags&MAIN_MENU_KDE && flags&MAIN_MENU_KDE_SUB) {
+		GtkWidget *pixmap = NULL;
+		if(need_separ)
+			add_menu_separator(root_menu);
+		need_separ = FALSE;
+		menu = create_kde_menu(NULL, fake_submenus, TRUE, TRUE);
+		if (g_file_exists("/usr/share/icons/exec.xpm")) {
+			pixmap = gnome_stock_pixmap_widget_at_size (NULL, "/usr/share/icons/exec.xpm",
+								    SMALL_ICON_SIZE,
+								    SMALL_ICON_SIZE);
+			if (pixmap)
+				gtk_widget_show (pixmap);
+		}
+		menuitem = gtk_menu_item_new ();
+		setup_menuitem (menuitem, pixmap, _("KDE menus"));
+		gtk_menu_append (GTK_MENU (root_menu), menuitem);
+		gtk_menu_item_set_submenu (GTK_MENU_ITEM (menuitem), menu);
+		gtk_signal_connect(GTK_OBJECT(menu),"show",
+				   GTK_SIGNAL_FUNC(rh_submenu_to_display),
+				   menuitem);
+		gtk_signal_connect(GTK_OBJECT(menu),"show",
+				   GTK_SIGNAL_FUNC(submenu_to_display),
+				   menuitem);
+	}
 	add_special_entries (root_menu, fake_submenus);
 	
 	return root_menu;
@@ -3365,6 +3428,8 @@
 	GtkWidget *user_sub = gtk_object_get_data(GTK_OBJECT(widget), "user_sub");
 	GtkWidget *redhat_off = gtk_object_get_data(GTK_OBJECT(widget), "redhat_off");
 	GtkWidget *redhat_sub = gtk_object_get_data(GTK_OBJECT(widget), "redhat_sub");
+ 	GtkWidget *kde_off = gtk_object_get_data(GTK_OBJECT(widget), "kde_off");
+ 	GtkWidget *kde_sub = gtk_object_get_data(GTK_OBJECT(widget), "kde_sub");
 	GtkWidget *debian_off = gtk_object_get_data(GTK_OBJECT(widget), "debian_off");
 	GtkWidget *debian_sub = gtk_object_get_data(GTK_OBJECT(widget), "debian_sub");
 	GtkWidget *pathentry = gtk_object_get_data(GTK_OBJECT(widget), "path");
@@ -3420,6 +3485,14 @@
 		menu->main_menu_flags |= MAIN_MENU_DEBIAN;
 		menu->main_menu_flags &=~ MAIN_MENU_DEBIAN_SUB;
 	}
+	if(GTK_TOGGLE_BUTTON(kde_off)->active)
+		menu->main_menu_flags &=~ (MAIN_MENU_KDE|MAIN_MENU_KDE_SUB);
+	else if(GTK_TOGGLE_BUTTON(kde_sub)->active)
+		menu->main_menu_flags |= MAIN_MENU_KDE|MAIN_MENU_KDE_SUB;
+	else {
+		menu->main_menu_flags |= MAIN_MENU_KDE;
+		menu->main_menu_flags &=~ MAIN_MENU_KDE_SUB;
+	}
 
 	if(menu->menu)
 		gtk_widget_destroy(menu->menu);	
@@ -3595,7 +3668,7 @@
 	gtk_object_set_data(GTK_OBJECT(dialog),"main_frame",f);
 	gtk_box_pack_start(GTK_BOX(vbox),f,FALSE,FALSE,0);
 	
-	table = gtk_table_new(3,4,FALSE);
+	table = gtk_table_new(5,4,FALSE);
 	gtk_container_set_border_width(GTK_CONTAINER(table),GNOME_PAD_SMALL);
 	gtk_container_add(GTK_CONTAINER(f),table);
 
@@ -3611,7 +3684,11 @@
 			      _("Red Hat menu (if found): "),"redhat",
 			      menu->main_menu_flags&MAIN_MENU_REDHAT,
 			      menu->main_menu_flags&MAIN_MENU_REDHAT_SUB);
-	add_menu_type_options(GTK_OBJECT(dialog),GTK_TABLE(table),3,
+ 	add_menu_type_options(GTK_OBJECT(dialog),GTK_TABLE(table),3,
+ 			      _("KDE menu (if found): "),"kde",
+ 			      menu->main_menu_flags&MAIN_MENU_KDE,
+ 			      menu->main_menu_flags&MAIN_MENU_KDE_SUB);
+	add_menu_type_options(GTK_OBJECT(dialog),GTK_TABLE(table),4,
 			      _("Debian menu (if found): "),"debian",
 			      menu->main_menu_flags&MAIN_MENU_DEBIAN,
 			      menu->main_menu_flags&MAIN_MENU_DEBIAN_SUB);
--- gnome-core-1.0.3/panel/menu.h.kdemenus	Mon Mar 15 00:05:39 1999
+++ gnome-core-1.0.3/panel/menu.h	Mon Mar 15 00:05:44 1999
@@ -19,8 +19,8 @@
 	MAIN_MENU_SYSTEM_SUB = 1<<3,
 	MAIN_MENU_REDHAT = 1<<4,
 	MAIN_MENU_REDHAT_SUB = 1<<5,
-	/*MAIN_MENU_KDE = 1<<6,
-	MAIN_MENU_KDE_SUB = 1<<7, KDE menus not done yet*/
+	MAIN_MENU_KDE = 1<<6,
+	MAIN_MENU_KDE_SUB = 1<<7,
 	MAIN_MENU_DEBIAN = 1<<8,
 	MAIN_MENU_DEBIAN_SUB = 1<<9
 };
