--- pmake/mk/bsd.lib.mk.rules	Tue Dec 16 12:52:33 1997
+++ pmake/mk/bsd.lib.mk	Tue Dec 16 12:53:52 1997
@@ -62,15 +62,13 @@
 lib${LIB}.a:: ${OBJS}
 	@echo building standard ${LIB} library
 	@rm -f lib${LIB}.a
-	@${AR} cTq lib${LIB}.a `lorder ${OBJS} | tsort` ${LDADD}
-	ranlib lib${LIB}.a
+	@${AR} csq lib${LIB}.a ${OBJS} ${LDADD}
 
 POBJS+=	${OBJS:.o=.po}
 lib${LIB}_p.a:: ${POBJS}
 	@echo building profiled ${LIB} library
 	@rm -f lib${LIB}_p.a
-	@${AR} cTq lib${LIB}_p.a `lorder ${POBJS} | tsort` ${LDADD}
-	ranlib lib${LIB}_p.a
+	@${AR} csq lib${LIB}_p.a ${POBJS} ${LDADD}
 
 llib-l${LIB}.ln: ${SRCS}
 	${LINT} -C${LIB} ${CFLAGS} ${.ALLSRC:M*.c}
