diff -Nur sendmail-8.9.1.orig/src/collect.c sendmail-8.9.1/src/collect.c
--- sendmail-8.9.1.orig/src/collect.c	Mon Jan 18 14:34:57 1999
+++ sendmail-8.9.1/src/collect.c	Mon Jan 18 14:42:00 1999
@@ -10,6 +10,7 @@
  *
  */
 
+#define MAXHDRZ 512 
 #ifndef lint
 static char sccsid[] = "@(#)collect.c	8.91 (Berkeley) 8/19/1998";
 #endif /* not lint */
@@ -65,6 +66,7 @@
 	HDR **hdrp;
 	register ENVELOPE *e;
 {
+	int hdrz = 0;
 	register FILE *volatile tf;
 	volatile bool ignrdot = smtpmode ? FALSE : IgnrDot;
 	volatile time_t dbto = smtpmode ? TimeOuts.to_datablock : 0;
@@ -330,7 +332,16 @@
 				mstate = MS_BODY;
 				goto nextstate;
 			}
-
+ 
+           if (hdrz++>MAXHDRZ)
+			{
+				sm_syslog(LOG_NOTICE, e->e_id,
+				"excessive headers from %s during message collect",
+				CurHostName ? CurHostName : "<local machine>");
+				errno = 0;
+				usrerr("451 Stop this. You are lame.");
+				goto readerr;
+			}
 			/* check for possible continuation line */
 			do
 			{
