--- sendmail-8.9.1/rmail/rmail.c.rmail	Mon Feb  9 14:30:13 1998
+++ sendmail-8.9.1/rmail/rmail.c	Mon Feb  9 14:30:42 1998
@@ -232,7 +232,6 @@
 	i = 0;
 	args[i++] = _PATH_SENDMAIL;	/* Build sendmail's argument list. */
 	args[i++] = "-oee";		/* No errors, just status. */
-	args[i++] = "-odq";		/* Queue it, don't try to deliver. */
 	args[i++] = "-oi";		/* Ignore '.' on a line by itself. */
 
 	/* set from system and protocol used */
