--- zip-2.1/zip.h.mike	Mon Dec 21 15:09:36 1998
+++ zip-2.1/zip.h	Mon Dec 21 15:10:40 1998
@@ -19,6 +19,7 @@
 
 /* Set up portability */
 #include "tailor.h"
+#include <strings.h>
 
 #define MIN_MATCH  3
 #define MAX_MATCH  258
@@ -277,12 +278,6 @@
 int setfileattr OF((char *, int));
 char *tempname OF((char *));
 int fcopy OF((FILE *, FILE *, ulg));
-
-#ifdef ZMEM
-   char *memset OF((char *, int, unsigned int));
-   char *memcpy OF((char *, char *, unsigned int));
-   int memcmp OF((char *, char *, unsigned int));
-#endif /* ZMEM */
 
         /* in system dependent fileio code (<system>.c) */
 #ifndef UTIL
--- zip-2.1/fileio.c.mike	Mon Dec 21 15:17:46 1998
+++ zip-2.1/fileio.c	Mon Dec 21 15:19:12 1998
@@ -704,65 +704,3 @@
 #endif /* NO_RENAME */
 
 
-#ifdef ZMEM
-
-/************************/
-/*  Function memset()  */
-/************************/
-
-/*
- * memset - for systems without it
- *  bill davidsen - March 1990
- */
-
-char *
-memset(buf, init, len)
-register char *buf;     /* buffer loc */
-register int init;      /* initializer */
-register unsigned int len;   /* length of the buffer */
-{
-    char *start;
-
-    start = buf;
-    while (len--) *(buf++) = init;
-    return(start);
-}
-
-
-/************************/
-/*  Function memcpy()  */
-/************************/
-
-char *
-memcpy(dst,src,len)           /* v2.0f */
-register char *dst, *src;
-register unsigned int len;
-{
-    char *start;
-
-    start = dst;
-    while (len--)
-        *dst++ = *src++;
-    return(start);
-}
-
-
-/************************/
-/*  Function memcmp()  */
-/************************/
-
-int
-memcmp(b1,b2,len)                     /* jpd@usl.edu -- 11/16/90 */
-register char *b1, *b2;
-register unsigned int len;
-{
-
-    if (len) do {             /* examine each byte (if any) */
-      if (*b1++ != *b2++)
-        return (*((uch *)b1-1) - *((uch *)b2-1));  /* exit when miscompare */
-       } while (--len);
-
-    return(0);        /* no miscompares, yield 0 result */
-}
-
-#endif  /* ZMEM */
