--- console-tools-1999.03.02/data/consolefonts/Makefile.am.fonts3	Wed Apr 14 12:29:07 1999
+++ console-tools-1999.03.02/data/consolefonts/Makefile.am	Wed Apr 14 12:33:56 1999
@@ -33,8 +33,9 @@
 iso08.f14.psf iso08.f16.psf iso09.f08.psf iso09.f14.psf iso09.f16.psf \
 iso14.f16.psf							\
 iso10.f08.psf iso10.f14.psf iso10.f16.psf koi8-14.psf koi8-8x14.psf \
-koi8-8x16.psf koi8-8x8.psf koi8b-8x16.psf koi8c-8x16.psf lat1-08.psf \
-lat1-10.psf lat1-12.psf lat1-14.psf lat1-16.psf lat1u-08.psf \
+koi8-8x16.psf koi8-8x8.psf koi8b-8x16.psf koi8c-8x16.psf \
+koi8u-8x8.psf koi8u-8x14.psf koi8u-8x16.psf \
+lat1-08.psf lat1-10.psf lat1-12.psf lat1-14.psf lat1-16.psf lat1u-08.psf \
 lat1u-10.psf lat1u-12.psf lat1u-14.psf lat1u-16.psf lat2-08.psf \
 lat2-10.psf lat2-12.psf lat2-14.psf lat2-16.psf lat4-08.psf lat4-10.psf \
 lat4-12.psf lat4-14.psf lat4-16+.psf lat4-16.psf lat4-19.psf \
--- console-tools-1999.03.02/data/consoletrans/Makefile.am.fonts3	Wed Apr 14 12:34:45 1999
+++ console-tools-1999.03.02/data/consoletrans/Makefile.am	Wed Apr 14 12:36:37 1999
@@ -57,7 +57,8 @@
 iso07.sfm iso08.sfm iso09.sfm iso10.sfm iso14.sfm iso15.sfm \
 ECMA144.sfm cp437.sfm def.sfm ethiopic.sfm koi8-r.sfm \
 lat1.sfm lat1u.sfm lat2u.sfm lat4.sfm lat4u.sfm \
-armscii8.sfm cp850.sfm cp850b.sfm tcvn.sfm viscii.sfm latin2-ucw.uni lat2u.uni
+armscii8.sfm cp850.sfm cp850b.sfm tcvn.sfm viscii.sfm latin2-ucw.uni \
+lat2u.uni koi8u2ruscii
 
 OLD_ACM_FILES = koi2alt null space trivial vga2iso zero \
 cp437_to_iso01.trans cp850_to_iso01.trans \
