--- console-tools-1999.03.02/kbdtools/Makefile.in.rootpart	Wed Mar 31 18:44:44 1999
+++ console-tools-1999.03.02/kbdtools/Makefile.in	Wed Mar 31 18:45:26 1999
@@ -127,7 +127,8 @@
 YFLAGS = -d
 LDADD = ../lib/libctlocal.a ../lib/libctutils.la ../lib/libconsole.la 
 
-loadkeys_LDADD = ../lib/libcfont.la $(LDADD) @LEXLIB@
+loadkeys_LDADD = ../lib/.libs/libcfont.a ../lib/libctlocal.a \
+ ../lib/.libs/libctutils.a ../lib/.libs/libconsole.a @LEXLIB@
 
 bin_SCRIPTS = mk_modmap
 
--- console-tools-1999.03.02/screenfonttools/Makefile.in.rootpart	Wed Mar 31 18:43:38 1999
+++ console-tools-1999.03.02/screenfonttools/Makefile.in	Wed Mar 31 18:44:32 1999
@@ -120,7 +120,7 @@
 bin_PROGRAMS = consolechars showcfont
 LDADD = ../lib/libctlocal.a ../lib/libconsole.la ../lib/libctutils.la
 
-consolechars_LDADD = ../lib/libctlocal.a ../lib/libconsole.la ../lib/libcfont.la ../lib/libctutils.la
+consolechars_LDADD = ../lib/libctlocal.a ../lib/.libs/libconsole.a ../lib/.libs/libcfont.a ../lib/.libs/libctutils.a
 
 bin_SCRIPTS = unicode_start unicode_stop charset
 
@@ -139,8 +139,8 @@
 LIBS = @LIBS@
 consolechars_SOURCES = consolechars.c
 consolechars_OBJECTS =  consolechars.o
-consolechars_DEPENDENCIES =  ../lib/libctlocal.a ../lib/libconsole.la \
-../lib/libcfont.la ../lib/libctutils.la
+consolechars_DEPENDENCIES =  ../lib/libctlocal.a ../lib/.libs/libconsole.a \
+../lib/.libs/libcfont.a ../lib/.libs/libctutils.a
 consolechars_LDFLAGS = 
 showcfont_SOURCES = showcfont.c
 showcfont_OBJECTS =  showcfont.o
