--- console-tools-1999.03.02/data/consolefonts/Makefile.am.jbj	Fri Apr  9 17:23:58 1999
+++ console-tools-1999.03.02/data/consolefonts/Makefile.am	Fri Apr  9 17:24:48 1999
@@ -49,4 +49,4 @@
 lat0-08.psf lat0-10.psf lat0-12.psf lat0-14.psf lat0-16.psf \
 t.psf t850b.psf tcvn8x16.psf viscii10-8x16.psf \
 lat0-sun16.psf lat2-sun16.psf ucw08.psf ucw11m.psf ucw11z.psf ucw16.psf \
-latin5-16.psf
+latin5-16.psf lat2u-16.psf RUSCII_8x14.psf RUSCII_8x8.psf RUSCII_8x16.psf
--- console-tools-1999.03.02/data/consoletrans/Makefile.am.jbj	Fri Apr  9 17:27:06 1999
+++ console-tools-1999.03.02/data/consoletrans/Makefile.am	Fri Apr  9 17:27:58 1999
@@ -57,7 +57,7 @@
 iso07.sfm iso08.sfm iso09.sfm iso10.sfm iso14.sfm iso15.sfm \
 ECMA144.sfm cp437.sfm def.sfm ethiopic.sfm koi8-r.sfm \
 lat1.sfm lat1u.sfm lat2u.sfm lat4.sfm lat4u.sfm \
-armscii8.sfm cp850.sfm cp850b.sfm tcvn.sfm viscii.sfm latin2-ucw.uni
+armscii8.sfm cp850.sfm cp850b.sfm tcvn.sfm viscii.sfm latin2-ucw.uni lat2u.uni
 
 OLD_ACM_FILES = koi2alt null space trivial vga2iso zero \
 cp437_to_iso01.trans cp850_to_iso01.trans \
